In [1]:
bw(s,msq,Γ)=1/(msq-s-1.0im*sqrt(msq)*Γ)

bw (generic function with 1 method)

In [2]:
Pkg.add("Plots")
Pkg.add("DataFrames")
Pkg.add("StatsBase")
Pkg.add("QuadGK")
Pkg.add("GR")

INFO: Package Plots is already installed
INFO: METADATA is out-of-date — you may not have the latest version of Plots
INFO: Use `Pkg.update()` to get the latest versions of your packages
INFO: Package DataFrames is already installed
INFO: METADATA is out-of-date — you may not have the latest version of DataFrames
INFO: Use `Pkg.update()` to get the latest versions of your packages
INFO: Package StatsBase is already installed
INFO: METADATA is out-of-date — you may not have the latest version of StatsBase
INFO: Use `Pkg.update()` to get the latest versions of your packages
INFO: Package QuadGK is already installed
INFO: METADATA is out-of-date — you may not have the latest version of QuadGK
INFO: Use `Pkg.update()` to get the latest versions of your packages
INFO: Package GR is already installed
INFO: METADATA is out-of-date — you may not have the latest version of GR
INFO: Use `Pkg.update()` to get the latest versions of your packages


In [2]:
using Plots
gr()

In [3]:
plot([
        x->real(bw(x,0.892^2,0.15)),
        x->imag(bw(x,0.892^2,0.15))
    ],
    linspace(0.4,1.3,100),
    lab = ["real", "imag"]
)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 1.2 
 
 
 -2 
 
 
 0 
 
 
 2 
 
 
 4 
 
 
 6 
 
<polyline clip-path="url(#clip9902)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 29.2591,180.71 34.8254,179.343 40.3916,177.934 45.9578,176.482 51.524,174.984 57.0902,173.441 62.6565,171.852 68.2227,170.214 73.7889,168.529 79.3551,166.796 
 84.9213,165.015 90.4876,163.187 96.0538,161.313 101.62,159.396 107.186,157.438 112.752,155.444 118.319,153.419 123.885,151.372 129.451,149.312 135.017,147.252 
 140.584,145.207 146.15,143.199 151.716,141.252 157.282,139.396 162.848,137.671 168.415,136.122 173.981,134.805 179.547,133.789 185.113,133.153 190.68,132.992 
 196.246,133.415 201.812,134.547 207.378,136.528 212.944,139.508 218.511,143.644 224.077,149.091 229.643,155.988 235.209,164.445 240.776,174.523 246.342,186.215 
 251.908,199.425 257.474,213.963 263.04,229.541 268.607,245.786 274.173,262.267 279.739,278.533 285.305,294.15 290.872,308.743 296.438,322.02 302.004,333.784 
 307.57,343.937 313.136,352.468 318.703,359.435 324.269,364.946 329.835,369.139 335.401,372.168 340.967,374.192 346.534,375.36 352.1,375.813 357.666,375.677 
 363.232,375.06 368.799,374.06 374.365,372.755 379.931,371.216 385.497,369.497 391.063,367.647 396.63,365.703 402.196,363.697 407.762,361.654 413.328,359.594 
 418.895,357.534 424.461,355.485 430.027,353.46 435.593,351.464 441.159,349.504 446.726,347.585 452.292,345.709 457.858,343.879 463.424,342.096 468.991,340.36 
 474.557,338.673 480.123,337.034 485.689,335.442 491.255,333.897 496.822,332.397 502.388,330.943 507.954,329.532 513.52,328.163 519.087,326.836 524.653,325.549 
 530.219,324.3 535.785,323.089 541.351,321.914 546.918,320.773 552.484,319.667 558.05,318.592 563.616,317.549 569.183,316.536 574.749,315.553 580.315,314.597 
 
 "/>
<polyline clip-path="url(#clip9902)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 29.2591,229.484 34.8254,228.425 40.3916,227.3 45.9578,226.103 51.524,224.83 57.0902,223.473 62.6565,222.025 68.2227,220.478 73.7889,218.825 79.3551,217.055 
 84.9213,215.159 90.4876,213.125 96.0538,210.941 101.62,208.592 107.186,206.065 112.752,203.341 118.319,200.403 123.885,197.231 129.451,193.803 135.017,190.095 
 140.584,186.082 146.15,181.736 151.716,177.028 157.282,171.927 162.848,166.402 168.415,160.42 173.981,153.951 179.547,146.965 185.113,139.441 190.68,131.362 
 196.246,122.724 201.812,113.539 207.378,103.845 212.944,93.7051 218.511,83.2224 224.077,72.5429 229.643,61.862 235.209,51.4268 240.776,41.5326 246.342,32.5122 
 251.908,24.7169 257.474,18.4891 263.04,14.1292 268.607,11.8626 274.173,11.811 279.739,13.9774 285.305,18.245 290.872,24.3929 296.438,32.1234 302.004,41.0951 
 307.57,50.9566 313.136,61.3737 318.703,72.049 324.269,82.7333 329.835,93.2287 335.401,103.387 340.967,113.103 346.534,122.312 352.1,130.976 357.666,139.081 
 363.232,146.631 368.799,153.64 374.365,160.133 379.931,166.136 385.497,171.682 391.063,176.802 396.63,181.527 402.196,185.889 407.762,189.917 413.328,193.638 
 418.895,197.079 424.461,200.262 430.027,203.21 435.593,205.944 441.159,208.48 446.726,210.837 452.292,213.028 457.858,215.069 463.424,216.971 468.991,218.746 
 474.557,220.404 480.123,221.956 485.689,223.408 491.255,224.769 496.822,226.047 502.388,227.246 507.954,228.375 513.52,229.437 519.087,230.437 524.653,231.381 
 530.219,232.272 535.785,233.113 541.351,233.909 546.918,234.663 552.484,235.377 558.05,236.053 563.616,236.696 569.183,237.306 574.749,237.885 580.315,238.437 
 
 "/>
 
 
 
 
 real 
 
 
 
 imag

## Integral over Dalitz plot

In [4]:
using QuadGK

In [187]:
# first, lambda
λ(x,y,z)=x^2+y^2+z^2-2*x*y-2*y*z-2*z*x
# second, integrate phase space 
function projectDalitzX(ampl,s1,s,m1sq,m2sq,m3sq)
    s1 < (sqrt(m2sq)+sqrt(m3sq))^2 && return 0
    s1 > (sqrt(s)-sqrt(m1sq))^2 && return 0
    ranges = m1sq+m2sq+(s-s1-m1sq)*(s1+m2sq-m3sq)/(2*s1)-sqrt(λ(s,s1,m1sq)*λ(s1,m2sq,m3sq))/(2*s1)*[-1.,1.]
    return quadgk(
        s3->abs(ampl(s1,s3,s,m1sq,m2sq,m3sq))^2,
        ranges[2],
        ranges[1])[1]/(8*π)^2/s;
end

projectDalitzX (generic function with 1 method)

In [188]:
# masses
mB = 5.279;
mJpsi = 3.686; # 3.01;
mπ = 0.139;
mK = 0.485;
mKs = 0.893;
ΓKs = 0.05;
mμ = 0.105;
# projection of the dalitz plot for the B to J/psi pi K
bw_amp(s1,s3,s,m1sq,m2sq,m3sq) = bw(s1,mKs^2,ΓKs)
phsp_s1(s1,s,m1sq,m2sq,m3sq) = sqrt(λ(s1,m2sq,m3sq)*λ(s,s1,m1sq))/s1/(8π)^2/s;
plot([
        e1->log(projectDalitzX(bw_amp,e1^2,mB^2,mJpsi^2,mπ^2,mK^2)),
        e1->log(abs(bw(e1^2,mKs^2,ΓKs))^2*phsp_s1(e1^2,mB^2,mJpsi^2,mπ^2,mK^2))
    ],
    linspace((mπ+mK)+0.01,(mB-mJpsi)-0.01,200))

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.8 
 
 
 1.0 
 
 
 1.2 
 
 
 1.4 
 
 
 -10 
 
 
 -8 
 
 
 -6 
 
 
 -4 
 
 
 -2 
 
<polyline clip-path="url(#clip1502)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.8488,244.957 38.5848,235.986 41.3208,229.217 44.0568,223.664 46.7928,218.878 49.5288,214.614 52.2648,210.725 55.0008,207.114 57.7369,203.716 60.4729,200.483 
 63.2089,197.377 65.9449,194.371 68.6809,191.443 71.4169,188.575 74.1529,185.752 76.8889,182.96 79.6249,180.188 82.361,177.427 85.097,174.667 87.833,171.9 
 90.569,169.118 93.305,166.315 96.041,163.482 98.777,160.613 101.513,157.701 104.249,154.738 106.985,151.718 109.721,148.633 112.457,145.475 115.193,142.236 
 117.929,138.908 120.665,135.48 123.401,131.944 126.137,128.288 128.873,124.5 131.609,120.568 134.345,116.476 137.081,112.211 139.817,107.754 142.553,103.087 
 145.289,98.1883 148.025,93.0369 150.761,87.6087 153.497,81.8793 156.233,75.8249 158.969,69.4252 161.705,62.6685 164.441,55.5603 167.177,48.1393 169.913,40.5032 
 172.649,32.8489 175.385,25.5257 178.121,19.0831 180.857,14.2572 183.593,11.811 186.329,12.2235 189.065,15.426 191.801,20.8434 194.537,27.6946 197.273,35.2786 
 200.009,43.0943 202.745,50.8323 205.481,58.3224 208.217,65.4833 210.953,72.2853 213.689,78.7275 216.425,84.8236 219.162,90.5947 221.898,96.0641 224.634,101.256 
 227.37,106.192 230.106,110.895 232.842,115.383 235.578,119.674 238.314,123.784 241.05,127.729 243.786,131.519 246.522,135.168 249.258,138.686 251.994,142.082 
 254.73,145.364 257.466,148.542 260.202,151.621 262.938,154.608 265.674,157.509 268.41,160.33 271.146,163.074 273.882,165.748 276.618,168.354 279.354,170.897 
 282.09,173.38 284.826,175.807 287.562,178.18 290.298,180.502 293.034,182.776 295.77,185.004 298.506,187.189 301.242,189.332 303.978,191.436 306.714,193.503 
 309.45,195.533 312.186,197.53 314.922,199.493 317.658,201.426 320.394,203.328 323.13,205.202 325.866,207.049 328.602,208.869 331.338,210.664 334.074,212.436 
 336.81,214.184 339.546,215.91 342.282,217.615 345.018,219.299 347.754,220.964 350.49,222.61 353.226,224.238 355.962,225.849 358.698,227.443 361.434,229.021 
 364.17,230.584 366.906,232.132 369.642,233.666 372.378,235.187 375.114,236.694 377.85,238.19 380.586,239.673 383.322,241.145 386.058,242.606 388.794,244.057 
 391.53,245.498 394.266,246.93 397.002,248.352 399.738,249.766 402.474,251.172 405.21,252.57 407.946,253.961 410.682,255.346 413.418,256.724 416.154,258.096 
 418.89,259.462 421.626,260.823 424.362,262.18 427.098,263.533 429.834,264.881 432.57,266.226 435.306,267.568 438.042,268.908 440.778,270.246 443.514,271.581 
 446.25,272.916 448.986,274.25 451.722,275.584 454.458,276.918 457.194,278.253 459.93,279.589 462.666,280.926 465.402,282.267 468.139,283.61 470.875,284.957 
 473.611,286.308 476.347,287.665 479.083,289.027 481.819,290.395 484.555,291.771 487.291,293.155 490.027,294.549 492.763,295.952 495.499,297.367 498.235,298.794 
 500.971,300.235 503.707,301.691 506.443,303.163 509.179,304.654 511.915,306.165 514.651,307.698 517.387,309.255 520.123,310.839 522.859,312.453 525.595,314.099 
 528.331,315.782 531.067,317.506 533.803,319.275 536.539,321.094 539.275,322.971 542.011,324.913 544.747,326.928 547.483,329.027 550.219,331.224 552.955,333.535 
 555.691,335.981 558.427,338.587 561.163,341.389 563.899,344.432 566.635,347.783 569.371,351.535 572.107,355.836 574.843,360.927 577.579,367.257 580.315,375.813 
 
 "/>
<polyline clip-path="url(#clip1502)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.8488,244.957 38.5848,235.986 41.3208,229.217 44.0568,223.664 46.7928,218.878 49.5288,214.614 52.2648,210.725 55.0008,207.114 57.7369,203.716 60.4729,200.483 
 63.2089,197.377 65.9449,194.371 68.6809,191.443 71.4169,188.575 74.1529,185.752 76.8889,182.96 79.6249,180.188 82.361,177.427 85.097,174.667 87.833,171.9 
 90.569,169.118 93.305,166.315 96.041,163.482 98.77

## Phase space MonteCarlo

In [189]:
# prepare array of weights
nbinsX = 300;
ms3_bins = linspace((mJpsi+mπ)^2,(mB-mK)^2,nbinsX+1)
density(s3) = sqrt(λ(mB^2,s3,mK^2)*λ(s3,mJpsi^2,mπ^2))/s3
weights = [density((ms3_bins[i]+ms3_bins[i+1])/2) for i in 1:nbinsX];

In [190]:
using StatsBase

In [191]:
nEvents = 1000000;
inds = StatsBase.sample(1:nbinsX, Weights(weights), nEvents)
s3_data = [ms3_bins[i]+rand()*(ms3_bins[i+1]-ms3_bins[i]) for i in inds];

In [192]:
histogram(s3_data, bins=200)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 16 
 
 
 18 
 
 
 20 
 
 
 22 
 
 
 0 
 
 
 2000 
 
 
 4000 
 
 
 6000 
 
 
 8000 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

In [193]:
using DataFrames

In [194]:
### Kinematic function
λ(x,y,z) = x^2+y^2+z^2-2*x*y-2*y*z-2*z*x;
density(s1,s,m1sq,m2sq,m3sq) = sqrt(λ(s,s1,m1sq)*λ(s1,m2sq,m3sq))/s1
s3_ranges(s1,s,m1sq,m2sq,m3sq) = m1sq+m2sq+(s-s1-m1sq)*(s1+m2sq-m3sq)/(2*s1)-sqrt(λ(s,s1,m1sq)*λ(s1,m2sq,m3sq))/(2*s1)*[1.,-1.]

### Functions to boost and rotate
Rz(θ) = [cos(θ) -sin(θ) 0 ; sin(θ) cos(θ) 0 ; 0 0 1];
Ry(θ) = [cos(θ) 0 sin(θ); 0 1 0 ; -sin(θ) 0 cos(θ)];
function Λz(γ)
    β = sqrt(1-1/γ^2);
    [γ 0 0 β*γ; 0 1 0 0; 0 0 1 0; β*γ 0 0 γ]
end
Rot4(R3) = [1 [0 0 0]; [0; 0; 0] R3];
boost_rotate(γ,θ,ϕ) = Rot4(Rz(ϕ)*Ry(θ))*Λz(γ);

# scalar products
mprod(P1,P2) = P1[1]*P2[1]-P1[2:4]⋅P2[2:4]
mnorm(P1) = mprod(P1,P1);

In [195]:
nEvents = 100000;

# data storage
tree = DataFrame(
    [Array{Float64,1} for i in 1:6],
    [:P1, :P2, :P3, :P4, :Q1, :Q2],
    nEvents);

# s-data
# prepare array of weights
nbinsX = 300;
ms_bins = linspace((mJpsi+mπ)^2,(mB-mK)^2,nbinsX+1)
density(s) = sqrt(λ(mB^2,s,mK^2)*λ(s,mJpsi^2,mπ^2))/s
weights = [density((ms_bins[i]+ms_bins[i+1])/2) for i in 1:nbinsX];
# generate nEvents-s-values
inds = StatsBase.sample(1:nbinsX, Weights(weights), nEvents)
s_data = [ms3_bins[i]+rand()*(ms3_bins[i+1]-ms3_bins[i]) for i in inds];

# main look to fill trees with four-vectors
mπsq = mπ^2;
for i in 1:nEvents

    s = s_data[i];
    
    # P2-frame
    cosT4 = 2*rand()-1.; sinT4 = sqrt(1-cosT4^2); θ4 = acos(cosT4);
    phi4 = π*(2*rand()-1.);
    p4 = sqrt(λ(mB^2, s, mK^2))/(2*mB);
    e4 = sqrt(p4^2+mK^2);  e13 = sqrt(p4^2+s);
    tree[i,:P2] = [mB,0,0,0];
    tree[i,:P4] = [e4,-p4*sinT4*cos(phi4),-p4*sinT4*sin(phi4),-p4*cosT4];

    # from P1-P3 rest frame
    γ13 = e13/sqrt(s);
    cosT1 = 2*rand()-1.; sinT1 = sqrt(1-cosT1^2); θ1 = acos(cosT1);
    phi1 = π*(2*rand()-1.);
    p1 = sqrt(λ(s, mJpsi^2, mπ^2)/(4*s));
    e1 = sqrt(p1^2+mJpsi^2); e3 = sqrt(p1^2+mπ^2);
    tree[i,:P1] = boost_rotate(γ13,θ4,phi4)*[e1,p1*sinT1*cos(phi1),p1*sinT1*sin(phi1),p1*cosT1];
    tree[i,:P3] = boost_rotate(γ13,θ4,phi4)*[e3,-p1*sinT1*cos(phi1),-p1*sinT1*sin(phi1),-p1*cosT1];
    
    # from Q1-Q2 rest frame
    γ1 = e1/mJpsi;
    cosTμ = 2*rand()-1.; sinTμ = sqrt(1-cosTμ^2);
    phiμ = π*(2*rand()-1.);
    pμ = sqrt(λ(mJpsi^2, mμ^2, mμ^2))/(2*mJpsi);
    eμ = sqrt(pμ^2+mμ^2);
    tree[i,:Q1] = boost_rotate(γ13,θ4,phi4)*boost_rotate(γ1,θ1,phi1)*[eμ, pμ*sinTμ*cos(phiμ), pμ*sinTμ*sin(phiμ), pμ*cosTμ];
    tree[i,:Q2] = boost_rotate(γ13,θ4,phi4)*boost_rotate(γ1,θ1,phi1)*[eμ,-pμ*sinTμ*cos(phiμ),-pμ*sinTμ*sin(phiμ),-pμ*cosTμ];    
end

In [196]:
tree[1,:]

1×6 DataFrames.DataFrame
│ Row │ P1                                       │ P2                     │
├─────┼──────────────────────────────────────────┼────────────────────────┤
│ 1   │ [3.84578, 0.471331, -0.417996, 0.898071] │ [5.279, 0.0, 0.0, 0.0] │

│ Row │ P3                                          │
├─────┼─────────────────────────────────────────────┤
│ 1   │ [0.329804, -0.219515, -0.155883, -0.130242] │

│ Row │ P4                                        │
├─────┼───────────────────────────────────────────┤
│ 1   │ [1.10342, -0.251815, 0.573879, -0.767829] │

│ Row │ Q1                                     │
├─────┼────────────────────────────────────────┤
│ 1   │ [1.85848, -0.560121, 1.23744, 1.26409] │

│ Row │ Q2                                     │
├─────┼────────────────────────────────────────┤
│ 1   │ [1.9873, 1.03145, -1.65543, -0.366021] │

In [241]:
# check if s-distribution is the same as input
histogram([
        [mnorm(tree[i,:P1]+tree[i,:P3]) for i=1:nEvents],
        s_data
        ],
     nbins=200,fillalpha=0.7)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 12 
 
 
 15 
 
 
 18 
 
 
 21 
 
 
 0 
 
 
 250 
 
 
 500 
 
 
 750 
 
 
 1000 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1 
 
 
 
 
 y2

In [197]:
# check energy conservation
histogram([tree[i,:P2][1]-tree[i,:P3][1]-tree[i,:P4][1]-tree[i,:Q1][1]-tree[i,:Q2][1] for i=1:nEvents],
    nbins=200,
    xlims=(-0.1,0.1),lab="eB-sum_e")
# check B-invariant mass
histogram!([mB^2 - mnorm(tree[i,:P4]+tree[i,:P3]+tree[i,:Q1]+tree[i,:Q2]) for i=1:nEvents],
    nbins=200,
xlims=(-0.1,0.1),lab="mB^2-|sumP|^2")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 -0.10 
 
 
 -0.05 
 
 
 0.00 
 
 
 0.05 
 
 
 0.10 
 
 
 0 
 
 
 10000 
 
 
 20000 
 
 
 30000 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 eB-sum_e 
 
 
 
 
 mB^2-|sumP|^2

In [199]:
# check dalitz plot
histogram2d(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    [mnorm(tree[i,:Q1]+tree[i,:Q2]+tree[i,:P3]) for i in 1:nEvents],
    bins=(100,100),
    xlabel="t [pi Jpsi] (GeV)",
    ylabel="s [pi K] (GeV)",
    title = "Jpsi pi K phase space Dalitz plot"
    )

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 16 
 
 
 18 
 
 
 20 
 
 
 22 
 
 
 Jpsi pi K phase space Dalitz plot 
 
 
 t [pi Jpsi] (GeV) 
 
 
 s [pi K] (GeV) 
 
 
<image width="476" height="329" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAdwAAAFJCAYAAAAxCJwFAAAgAElEQVR4nO29abwW5ZX1vYwQGQ4Y
PSKTiCBwoCFEAihiFDzyODSCQ9oBMWrSAZx9uuVJojEGE+PQr9jtHDVtIC1xStSAtpJGhBgRBUSJ
NBwUEGVSPBjhMNhA5/mQ3/u+7LVXuCt0Uk7r/20frrqHqququGtda+09YMxfgP2bD/jDznXVHtVp
TM89Dgr12C+uTmOufqVVqG8e8Faon1jaJW0zd314a9S03DONadHov0P9+Lr3Q91jr/x5h3f4INTf
XPJCGtP3c4NDvWiPBaHevGN92qZh6zL6y/Y90iBjzKeOz33UH8AYY4z5LOAbrjHGGFMCvuEaY4wx
JWDt6DMOa68AMOTzh4Z64/Ydabvzu9XHMdv2CnXDtsZpmz5tV4b6N28dlMaMOPK3oZ4xr2+oWzT+
MG1TtVf+WxrTZEuo137whYrbdGq9NtTL32mTxvTsHPXY62cOCPXGbfl1rzo86rxbtjZJY3jfzKn/
fKhnb1uatmn4Qzwm726a7fPbmI8R/oVrjDHGlIBvuMYYY0wJ+IZrjDHGlIA1nk8RVU26BT226+f6
pzEnt9on1Kf1fC2NmbU8el03bmuUxqzaEjXFiwfODjVrmQBwdJsP0t+YyW/vHeqRnd8NdZu9f5+2
eaCuW6jbN/2vNOa4botDPXVJ94qf5cv7vxPql99tnca0aLw91CcOmJ3GMHdPHxy3OfiNNIa/07CO
b4f69fezb3jV5qgFb9ye/z/NfuS6DXnM5C1Ph7ph6xJfJ4z5C+BfuMYYY0wJ+IZrjDHGlIBvuMYY
Y0wJWJv5hDCo2ejkl+1eFTW7239wT6iv+OHo9DojapaEun3HlWnMXlWbQj3hqePSmKMOfDPU4+Yc
HOp+++apxZnHNx65MI15ZGGvUK/eEv9P2L1l1mfbNose2zWbm6YxrDmzljm4/aq0DdPqC++nv/Hn
PffQl0K9YlX7tA17gl9/L2vDzNz6qG33q66sh3fd7530N36vIn5p3oY/i+KmVbf62mIM4V+4xhhj
TAn4hmuMMcaUgG+4xhhjTAlYZ/kY0GjP6qTPnt5yZKjbNcvbsQ7JnlqV0duwNeqbM4TGyK8zckYe
8y+HbM4faCeUZ7WI1jr9ndjLlnvmTlnRIW3DvtvFGz6fxozuFbXr+WsOCHVV4xx63LZqQ/y8DS3T
mN4HrAj1gpUdQ/3s2qx3sg7N+rKibkPMsx7XP2cpc6/gmr03pjGD+86L28zOfmmm6z4xo1nNGfVe
DOvF1729Jv47ZUEDzoM2ny78C9cYY4wpAd9wjTHGmBLwDdcYY4wpAesjHwGdq4YGzfbc6k5pDOfg
qnzgbpQr3Ld31F45sxfIWtuAmkVpzO2zoq7H3l0F64cK9tS2a/rfaUy//d4L9eS39w91/+qsHX91
yDOhnjrzqDSG9eJT+rwc6sfmfzltw/nLqh8u9+K9e1HUmPtX5+PG+csKzq/eQLXKSf7uqVNCrfzT
/DrfvnBiGnPXvXH9QJG86PXv7BfqfVu/l8asWhF1c+5J/OzaeKwVc97fmv72/Ie/DPX2HfW+rpmP
Jf6Fa4wxxpSAb7jGGGNMCfiGa4wxxpSAtY6/MNyTFgCGNz0+1GN6xN6mylvKKA2Xfa3ssXzgua+k
bdjnumFH9p/23yf6d9V7MyOO/G2oZ8zrm8bUfdAi1N+9IeuHU28aGmrWSFWeMfe2ZX0WAGbX9Qg1
a9cq87jhw71CvYQ0RyDr6HPfi1omZysDwMJlnUPdqfXaNIY91E2bRO2y7cHR/wsAs2YdVvHzFtGP
WaOdtyDmRbOuDgCX9VsQavYjF4HnLwDU1XUNtfJCsz4/pz76mrm/L+Aev+ajwb9wjTHGmBLwDdcY
Y4wpAd9wjTHGmBKwjvEXZl7t8UnDrdQrdkztjPQ6o34xKNScKQzoTNvwPvtlL+TGbVGXPHHYU2nM
E1NOCDVnCgPA6+9Xh3r62hj2PLLzu2mbNqR3cq4zkDU61n2LaKLKzzmsY9TNK70PkD3LakxL0kTZ
u6u01ovuHxbq64ZkX+vmTXF/sgdY6bNf3j/2v2UNWsHzAQAmvx3zn3nNAR97AFi1OWrOJx78Rhoz
fn63UN9xdvQNr1madV/W57kPM6Dn0c6oNRI8H9g/DQA/f/9O+st2Xy/N/wj/wjXGGGNKwDdcY4wx
pgR8wzXGGGNKwJrEn8GgZqOTPsuc3K7y/2HYJ6h63XZvGTODWSMDgOnroj/2gi7Rs6pyklnvVLCX
tH2znF/LPVJZL2S9FgDueS1qeOzdBIBrX+gdas4i5ixgIOuF3OsWyP1ued8UyVJW/YWrq9eHeuJL
h4aaewsrHlnYK/2N9/niDXGSbMz26dTz9zdvHZTGcO9d7qkMAJcOfzLUH6xtFWrO2gaAo9tEzZ7n
EJCPE38+pc+qY8n0abuy4hiGc5wV/B1Y0790aV3axv17za7wL1xjjDGmBHzDNcYYY0rAN1xjjDGm
BLIgZv4/Gu1ZHTRbpc+yHrs4W1ZxdJsPQl1L3s0hvbKWyVor90cFgNpW8b2fXRs1xhaNs16rfJfM
4ParQq38nOxj7brfO2kMwxrjut/vk8awZst50XPIIwoAJ1KtND3WWjk7mbVNIPd05X6uQNYz6zbs
iJ9N6L7cO5g1UwD45bRjdvm6Izs1pG0eqIsaOeuqfyRql+p7L5h3SKjZ7z322OlpG+69zL2PgazH
nnfC1FDzdwaAX62sPF8Xb4jfu0j2N+vz7PcFdM/hnXnpb7P+PfGli8M144735qYx727iv9nf+1nB
v3CNMcaYEvAN1xhjjCkB33CNMcaYErCGuxP7Nx8Q9JdTmkdP5ZxoPQWQM4OV/5Q9f9y/k3u1AlkT
5b6wAFDVJOqb7GtUmcKs601alscMp1hZ5edlDXR3fI2cQwzkPODTXl0T6iGfj8cEAF5/r3WoVfZz
+6roa/75G/FLsr4MAE/Mjvose3mBrGeyzqtyfueuj3buqTOPSmPaNosa7W3HR51f+YYvHhgzmfdu
sy6NWUM66QjhfVX+3Z1RvY5Z7zyry9tpDGcas1as9m+LRtF/fN93fprGvPirIaHmrGfelwDQrPlm
GrMljWGd95a50SPedb990zaM0nln140M9V1v7BUmxMzN91jT/ZTiX7jGGGNMCfiGa4wxxpSAb7jG
GGNMCXyGtYJGKRe5qkn0rZ5ffXyoVe4sZ+WqXNzLz3441OPvPz3Uqm8ta6s3n5y9j4zS1pjeB8T+
rMp/yN5M7h0KAC+/S7opaWCqdyz7e1VfVdbbuO8r62pA7hX74yX5dVU/4Z1hXzGQ95U6tnUb4v9Z
2UfcTWj6rK0r3+iII38bataTTxyQe+je9OvaUPer/iCNYdRx4vnI6xJUvvHATjEnmTOxgdxXd+A5
sRfzr+89OW3TqfXaUM9a3iWN4VxvnldKG2adnz8LAEy85ew/632AnHmuel1//8l4nDhTnLViAFjc
ELO1rfN+MvEvXGOMMaYEfMM1xhhjSuAz81iCYxr/vnpkGtO/OloFOHJRtbbjR17qES0/bmXLD0f9
Afnx69cX5MeOt9XEx4EDB74Y6qqh2Xbz6+8

In [200]:
using GSL

In [201]:
# Clebsches and d-function
function ClebschGordon(j1,m1,j2,m2,j,m)
    factor = (j1+j2-m)%2==1 ? -1 : 1
    factor*sqrt(2*j+1)*sf_coupling_3j(2*j1,2*j2,2*j,2*m1,2*m2,-2*m)
end

function Wignerd(j,m1,m2,cosθ)
#    print("WignerD(",j,",",m1,",",m2,",",cosθ,")")
    (m1 < 0) && print("Error: Something is wrong with Wignerdhat!")
    (m2 == 0) && sqrt(4*π/(2*j+1))*sf_legendre_sphPlm(j,m1,cosθ)
end

function Wignerdhat(j,m1,m2,cosθ)
#    print("WignerD(",j,",",m1,",",m2,",",cosθ,")")
    (m1 < 0) && print("Error: Something is wrong with Wignerdhat!")
    (m2 == 0) && sqrt(4*π/(2*j+1))*sf_legendre_sphPlm(j,m1,cosθ)/sqrt(1-cosθ*cosθ)^(m1)
end


Wignerdhat (generic function with 1 method)

In [212]:
# tensor pre-factors
function Cs(p1,p2,p3,p4)
    s = mnorm(p3+p4)
    p3-p4 - (mnorm(p3)-mnorm(p4))/s*(p3+p4)
end
Bs(p1,p2,p3,p4) = p3+p4
function Ds(p1,p2,p3,p4)
    # minus is removed due to the crossing
    # i is removed to make amplitude real
    [-sum((σ-ν)*(σ-ρ)*(σ-μ)*(ν-ρ)*(ν-μ)*(ρ-μ)*
            p1[σ]*p2[ν]*p3[ρ]
        for σ=1:4, ν=1:4, ρ=1:4)
        for μ=1:4]
end

# kinematical functions
function ζs(j,l,s,zs)
    (l < j-1 || l > j+1) && return 0;
    # three cases
    (l==j) && return 0;
    ps = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    qs = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    (l==j-1) && return (ps*qs)^(j-1)/(4*π)*sqrt((2*j+1)*(2*l+1)/2.0)*
        ClebschGordon(j-1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)
    (l==j+1) && return λ(s,mJpsi^2,mB^2)*(ps*qs)^(j-1)/(4*π)*sqrt((2*j+1)*(2*l+1)/2.0)*
        ClebschGordon(j+1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)
end
function βs(j,l,s,zs)
    (l < j-1 || l > j+1) && return 0;
    # three cases
    (l == j) && return 0;
    ps = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    qs = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    (l == j-1) && return (ps*qs)^(j)/λ(s,mJpsi^2,mB^2)*
        (s+mJpsi^2-mB^2)/sqrt(2)*
        4/(4*π)*sqrt((2*j+1)*(2*l+1)/2.0)*(
            ClebschGordon(j-1,0,1,0,j,0)*Wignerdhat(j,0,0,zs)/sqrt(2)+
            ClebschGordon(j-1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)*zs
        )
    (l == j+1) && return (ps*qs)^(j)*
        (s+mJpsi^2-mB^2)*
        4/(4*π)*sqrt((2*j+1)*(2*j+3)/2.0)*(
            ClebschGordon(j+1,0,1,0,j,0)*Wignerdhat(j,0,0,zs)/sqrt(2)+
            ClebschGordon(j+1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)*zs
        )
end
function δs(j,l,s,zs)
    (l < j-1 || l > j+1) && return 0;
    # three cases
    (l == j-1 || l == j+1) && return 0;
    ps = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    qs = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    (l == j) && return (ps*qs)^(j-1)*sqrt(2.0)/(4*π)*(2*j+1)*Wignerdhat(j,1,0,zs)
end

function Zjl(j,l,p1,p2,p3,p4)
    s = mnorm(p3+p4)
    t = mnorm(p1+p3)
    u = mB^2+mJpsi^2+mπ^2+mK^2-s-t
    zs = (s*(t-u)+(mπ^2-mK^2)*(mJpsi^2-mB^2))/sqrt(λ(s,mB^2,mJpsi^2)*λ(s,mπ^2,mK^2))
#    print("zs = ",zs,"\n")
    Cs(p1,p2,p3,p4)*ζs(j,l,s,zs)+
        Bs(p1,p2,p3,p4)*βs(j,l,s,zs)+
        Ds(p1,p2,p3,p4)*δs(j,l,s,zs)
end

Zjl (generic function with 1 method)

In [213]:
# definition of the amplitude and square of matrix element
# i.e. contracted with lepton current
function ampl(p1,p2,p3,p4,pars)
    amp = [0,0,0,0]
    for j in 1:length(pars)
        ps = pars[j]
        (length(ps) != 3) && continue
        amp += (ps[1] == 0) ? 0 : ps[1]*Zjl(j,j-1,p1,p2,p3,p4);
        amp += (ps[2] == 0) ? 0 : ps[2]*Zjl(j,j,  p1,p2,p3,p4);
        amp += (ps[3] == 0) ? 0 : ps[3]*Zjl(j,j+1,p1,p2,p3,p4);
    end
    amp
end

function amplsq(p1,p2,p3,p4,q1,q2,pars)    
    amp = ampl(p1,p2,p3,p4,pars)
    ampc = conj(amp);
    real(mprod(amp,q1)*mprod(ampc,q2)+mprod(amp,q2)*mprod(ampc,q1)-mprod(amp,ampc)*mJpsi^2/2)
#    real(amp[2:4]⋅ampc[2:4])
end

amplsq (generic function with 1 method)

In [214]:
Zjl(5,6,tree[2,:P1],tree[2,:P2],tree[2,:P3],tree[2,:P4])

4-element Array{Float64,1}:
 -9.3423 
  7.17208
  4.94579
 11.8943 

In [216]:
hists = histogram(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    weights = [amplsq(tree[i,:P1],tree[i,:P2],
    tree[i,:P3],tree[i,:P4],tree[i,:Q1],tree[i,:Q2],
            [[1,0,0]]) for i=1:nEvents],
    nbins=100,norm=true, lab="j=1, l=0"
    )
histogram!(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    weights = [amplsq(tree[i,:P1],tree[i,:P2],
            tree[i,:P3],tree[i,:P4],tree[i,:Q1],tree[i,:Q2],
            [[0,0,1]]) for i=1:nEvents],
    nbins=100,norm=true, lab="j=1, l=2",fillalpha=0.4
    )

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.00 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 j=1, l=0 
 
 
 
 
 j=1, l=2

In [267]:
histogram2d(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    [mnorm(tree[i,:P1]+tree[i,:P3]) for i in 1:nEvents],
    weights = [amplsq(tree[i,:P1],tree[i,:P2],
            tree[i,:P3],tree[i,:P4],tree[i,:Q1],tree[i,:Q2],
    [[0,0,1]]) for i=1:nEvents],
#    weights = [abs(bw(mnorm(tree[i,:P3]+tree[i,:P4]),1.4^2,0.3))^2 for i=1:nEvents],
    nbins=50
    )

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 1 
 
 
 2 
 
 
 3 
 
 
 4 
 
 
 5 
 
 
 12 
 
 
 15 
 
 
 18 
 
 
 21 
 
 
<image width="491" height="364" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAesAAAFsCAYAAADsRpHBAAAXeElEQVR4nO3dS4xsiX0W8HOqq7v6
3bf73rl37p2nx8PEmDjYRo4DWbDIwskSsUqQIpQFIlJgwwoRFllFrGDPFgkJFpHYRIBQgDjEAU1s
jYcMmfHM3HncV99Hd9/uqq7nOWzIDqHvymPmb/v3W386VXXqVH1Vm/O1DT+0v779G32S+/WXhtHx
fuPv/asot/Zidrx+PI9y01/+1Si3+e/+dXa8D65GuT/99s9HufdPj6LcHx3vZMebTKPc2+33otzJ
5Z9HuVV31kZBgP9j8Hk/AQDg/01ZA0BxyhoAilPWAFCcsgaA4pQ1ABSnrAGgOGUNAMUpawAoTlkD
QHHKGgCKU9YAUJyyBoDifirXf17d/Va0kvX7fzNbZXr9H92JcqujG1Fu+P3vRrnZe9m61Oj1iyi3
vJuteB1/92ei3P0H16Nc9GY0TfOFL96Ocu+8kz2/K7vnUe6Dh9n7tj/Krpf/dOdWlPsfj5dR7v3B
R1Hu7vztKDedf/xT+b0AlflnDQDFKWsAKE5ZA0BxyhoAilPWAFCcsgaA4pQ1ABSnrAGgOGUNAMUp
awAoTlkDQHHKGgCK+4m6Yf9f2/71aBPiO//830THa28dRLnHv3ctyh194/0od/+/fjnKnZ1ciXJt
m01ldH12OVy/9SDKffTBK1Fuf3cc5dY35lHuD975K1FutlqLcs9vZ8/v+HI7yl3fmkS5x9OtKPfB
xWaUm3fZ+/vHp9nwSxdOsHx/+Z+j3OUiG8Tp+/lP1PcWJPyzBoDilDUAFKesAaA4ZQ0AxSlrAChO
WQNAccoaAIpT1gBQnLIGgOKUNQAUp6wBoDhlDQDFKWsAKO7HYr3maPtr0bzPvd95Nzre2pf2otx3
/tk3o9zXf+mPotz4XrbO9fb3s9WtX/iVP4hyD//nF6PcvQc3olyfjS01w7VVlJvMstWodx4/F+Ve
OziJcu+dHkW5abjOtb++iHLni2GUezJfj3Kr8P14Mss+7oPwWyF82GbRZbm7l8so92b/vSh3vrwf
5S7n6dpXeALhR8A/awAoTlkDQHHKGgCKU9YAUJyyBoDilDUAFKesAaA4ZQ0AxSlrAChOWQNAccoa
AIpT1gBQ3Od6Y/q23Yy2AN761t+Ijre/fx7ltg+yXBcOOCxnG1GuHWSLBv/trZ+Lctd2LqLcg4v9
KJcOVpyFAxNfvf4gyg0H2eDHn9x7IcpdG82i3O3xdpR7bW8c5R5cbkW5h9NsyGMZLmU8N8rO33H4
uO+eZ9fptVH2W3+aPb3mZJENeaRDMg/6p1HudPA4yl30T6LcyfT9KLdanUa5vlkaEME/awCoTlkD
QHHKGgCKU9YAUJyyBoDilDUAFKesAaA4ZQ0AxSlrAChOWQNAccoaAIpT1gBQnLIGgOI+1zWXXzv8
h9F+zm995cPoeC+/+GmU++ROtt60s3kZ5SbTbG3p3UfPRbmNtWz16PF0FOVuj7Pc3379gyj39vHN
KPfxeDPKXRstotwnk+x1XGTjTc3PHmTrXCfzbK1qOMjmoM4X2bpZahyOMp3Ms+Olq1Zt+O0xW2UH
fDTProNVkx3vrJlEub7JPm/jQXa8yyZb+7pYPYpy5/M7UW7VZY/b93MrXj+G/LMGgOKUNQAUp6wB
oDhlDQDFKWsAKE5ZA0BxyhoAilPWAFCcsgaA4pQ1ABSnrAGgOGUNAMV9rjd0/72v/Z3ojvw39rIb
1N8/P4hyh5vZDfm/fS8brPjK4WmUu32+F+XmXfYbatlnb9+ru+Mol5oss2GL6SobrEhf73iZ5c7m
2eOO1rJBiJd2plHug/NsuGSyyt6303B4Yzd7O5rd9Sz3/nk2bLEeLnmM1rLcZJm9H0+X2VLLrMly
F032/o6a7ATeHWSDQunX76Q/iXLT1XmUmy2z76vFKnvcvp8aBvn/wD9rAChOWQNAccoaAIpT1gBQ
nLIGgOKUNQAUp6wBoDhlDQDFKWsAKE5ZA0BxyhoAilPWAFCcsgaA4sK9nmfz6u4vR/M5VzYvo+N9
+85LUe4Lu9nqzJ88eD7KnYbrTafzUZT70ycbUW4t/Am1H64ofTzej3I3t7K1pTuT7Ane2MpWlM4W
2WjPTriSla5pZa+2ad4524pyRxvZEc/D17udXX7NlY3s9a7ClbZhuKa1t57lzsM1ra7Jcocb2dfW
SbhaNg7Xr9owd7W7kT1um63hzdvse3IQ5rbXr0W5yza7AJfdRfTGdV22dtj3cyte/xf+WQNAccoa
AIpT1gBQnLIGgOKUNQAUp6wBoDhlDQDFKWsAKE5ZA0BxyhoAilPWAFCcsgaA4n4kN0z/Jy/9g+jG
7vvrq+h4O8Msd2eSDWXcvcxe9lF2uOZimeWe28yGCu6Hz+8yOy3NbjjXcjTKnt9x+PzSoYzr4XlJ
L9ZBGLwzyYLXN7PjPV1kuXAno9kMf0pPw+vgbJGd551h9gQ/mWQveC18wc9vZhfqJ5fZQsekyXLp
QMeqz67op4NsUGjRZs9v1WRfMItmGuUm3WmUW/bZ8RarbJBk2WXHW3UXUa7vZlmuWf5EDIP4Zw0A
xSlrAChOWQNAccoaAIpT1gBQnLIGgOKUNQAUp6wBoDhlDQDFKWsAKE5ZA0BxyhoAilPWAFBcuMf0
bE7n2cjJ+iD7rfDO2Wf7ND8NV3sez9ai3Eb4Osbh+Mt2+HJH4U+t03m2trQezlUdjrLHHYdrZKnJ
Knt+F+H6VbpGdpyNBTXhWFUzDefI5uH7+2SezW4dbWTXc7rmdriencB5n11/T+bZiVlvsxOz22cX
aroKdqd9EuX2+70od9Jm61frfTb/14avY9Fma1WDJrteukF2wbRtdry+z47XD8Ln10+jC7Dvw+L6
nPhnDQDFKWsAKE5ZA0BxyhoAilPWAFCcsgaA4pQ1ABSnrAGgOGUNAMUpawAoTlkDQHHKGgCKe6aF
jLXBXnRD9Fk4VPBfHmaLC4smu7H7KHw5m+EN4PsmGyBIc5erLDcJhxTWmuy+8/sbWe44u999czTK
jhfuNzT3L7PjjbK3rXkwyxZE9ofZAV/Pdhni15EOiIS7Fs1e+DrCt6N5usguwK217Lf+ssseeX89
ex2zLjvP01V2AtPP76vt1Sj30Sob/NjsN6Pcssmu53R4Y7c9inKTcGhk2WfDIE0bLgCFn4/lapIF
w+ulCz8ifR9+MJtwuSnknzUAFKesAaA4ZQ0AxSlrAChOWQNAccoaAIpT1gBQnLIGgOKUNQAUp6wB
oDhlDQDFKWsAKE5ZA0Bxz7S6tTt6JcodT7NVknQ1atRuRLk/b29Hude6l6Pczlp2eu4sxlFuvc1+
G90YbkW5QbjpchrON+2vZ89vmo0ANSfhetOVcG3pbJ6tI/3MXva+PQ7Hgk7mWe76Zvb8xqtwnSt8
f0/C15Gulo3CC2sVzqrthe9v6mKZXVdd+PzW2uz1Puyyz3kbfq8N++w6HQ/Oo1y6HtaFK4bDJlvJ
asPvtfS8tOF/yOEg+57s+nS1LPue7JrsC6FP35Bwncs/awAoTlkDQHHKGgCKU9YAUJyyBoDilDUA
FKesAaA4ZQ0AxSlrAChOWQNAccoaAIpT1gBQ3DMNeeysXYtyx6tJlDttT6PcrL2Mcodd9vxmTXZj
9zvdwyi30+5Eud0mGyQ5XS6i3JVhdrx0oGOR3ce+GS+z4PVwOWIY/

## Analytical approach
Now, I get the same distributions analytically
 - Matrix element square is lorethz invariant
 - in s-channel CMS, amplitude is straightforward
 - integrals over phase space are analytical as long as amplitude depend on (s, sz)
 - avaraging over J/psi polarization is a little bit tricky (see comment in the cell below)

In [206]:
function amplsp_analytic(s,zs,pars)
    # states along z axis
    p = sqrt(λ(s,mB^2,mJpsi^2)/(4*s))
    E1 = sqrt(p^2+mJpsi^2)
    E2 = sqrt(p^2+mB^2)
    p1 = [E1, 0,0,p]
    p2 = [E2, 0,0,p]
    # scattered states
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    E3 = sqrt(q^2+mπ^2)
    E4 = sqrt(q^2+mK^2)
    sinθ = sqrt(1-zs^2)
    p3 = [E3,-q*sinθ,0,-q*zs]
    p4 = [E4, q*sinθ,0,q*zs]
    # amplitude
    amp = ampl(p1,p2,p3,p4,pars)
    ampc = conj(amp)
    # avaraging over muon angles is easy in J/psi rest frame
    # the answer there is \vec A \cdot \vec A (some numerical factor)
    # therefore, I have to find \vec A^2 in Jpsi rest frame
    # I do it using scalar product of A to p1
    vecTildeAsq = mprod(amp,p1)*mprod(p1,ampc)/mJpsi^2 - mprod(amp,ampc)
    # attach muon current avaraged over muons angles
    ksq = λ(mJpsi,mμ^2,mμ^2)/(4*mJpsi^2)
    return 2*vecTildeAsq*(mJpsi^2/4-ksq/3)
end

amplsp_analytic (generic function with 1 method)

In [207]:
plot(0.5:0.03:2.5, -.99:0.03:0.99, 
    (s,zs)->(sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s*
amplsp_analytic(s,zs,[[0,0,1]])), t=:heatmap)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 -1.0 
 
 
 -0.5 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
<image width="475" height="364" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAdsAAAFsCAYAAACEtRP5AAAgAElEQVR4nO3dy3Jk2ZFeYT+XCACZ
WUU2KZGSKGuapJkeTU/SL9qSWZs11azMBBC3ExrANBJ97aIDHgWU1jf12ieugONY/uU+/dN//R/X
AP/9t/+W1v7bH/9XWvsPf/7ntPb5v+Tn1j9PaW37039Oa6c//Dk/95t/TGvXL39Ka/v979PasnxO
axER07zmxWv+lm/bMa1dLt/T2vn0U/5wh39Na9PTv6S1+ftf0tryLa/NP0Htr/n3KSIi/vo1LW1/
zY9dvt6ntfNPn9La8af8c3z+mteevufXfHp6SGuPz/nz/Ea1411ai4j4dtrnj3na5efOS1r7jrUZ
avnP8OMlLcXTGWqX/Gfmacsv+rTBRSPiKU5p7TkOeW16TmuH6SmtHSOvna75Nc9Qu1zhdwb8Ptnw
XP6+vJzN39drwHt+zT+ra2xwLq/hObwmtj54vNq5/CdGkiS9CZutJEnNbLaSJDWz2UqS1MxmK0lS
s/UJUoUREd8hyUjJyqefvqS1/V9+SGvzPSRZ5/+Z1nbnPAF3eYKE65c8GX26/zGvrXkaNSLiOufv
W0CabbrkCcE450nG6ZgnlZdD/vrnR6h9y+O/07c8/Tx9/ZbWrl8HKcev+ffx8o0Sx3ly+Pgt/6wO
UHt6zFPFT0/5c3l8zpPD3w957bGYKI6IeD7n6fenS/6ePhdrhy1PHB8gAHqENPJpy38uTpBGpXNn
TKpGnCN/QucJfqdAbYM0LtWulNSlNG4xqXvFNO4ocVtL5Nav2fF45O0fzztbSZKa2WwlSWpms5Uk
qZnNVpKkZuvTJQ86REQ8XfLgxbfnPEDyHUbd3f3vPHi03OUBoXXOgzfzOQ86TU95eOj6JR9leL3P
X8O25/F5Meej7sh0yQMU0yl/b6YjhKeeofaYvzfxmJ+7focxcPnHFNtjHgKKiDh/y79TVDt+z2sH
qD1TCIoCgAcISFEICsYuUkDqGX4OIyKeLvn3jWrlEBQEnY6QSTpB7uQEgR0OQeVP5gS1l7MQgoKz
FzwHASkcV1gbZdgx5nA8khDqrwpeVdw6PFXjna0kSc1stpIkNbPZSpLUzGYrSVIzm60kSc3W50Ea
+TuMgaOF1o+wQPv+r/kox2Wfj/O7n/PU2Xp+TGvTU544nj/n4wrjPn99847H58Vc/DsGFmFPJ1jK
fIQxiIc8OXx9hkRi/pbG9py//u0xf9/O3/MUb0TE6TGvnyA5fIBzzzBa8RmSw0/VVDHUcPwpjlzk
dHs1cfwMieNnCPLSSMYT1vKf4WN1XCONXBykkWnsYn0kYy1VvBUTxzyusZoMfs24xryGV/0YoeIy
72wlSWpms5UkqZnNVpKkZjZbSZKarRSCiIh4POfBi2/HfLzc16d8T+jdVxjJuINxhfQP7/A8l8Nz
WpvhuUwwOjIG+ajynzE0eQ3yUVfIR12P+ZO5QgiIQlAX2Nl6fsprJ6hFRBwhzHSEs88wPvFwyL+n
TxBmeoLvd3X37NMZahByop/Dl7MUdKrVqntpKejUEoKCc7STNoIDVPWRjMVxjR07a4v7bHnX7cuV
a97Tztrb8s5WkqRmNltJkprZbCVJamazlSSpmc1WkqRmK0zri4iIRxj19gjJym+QDn2AUY7LT4N4
dGKDtOYOUqzLwyGtzTA6cloHbxyMlkQwPu8KnwWlsbdjPgZwO+af4RlSvOdnqMG5I6SYIyKOcPYA
6WCqPVPiuCNVXBy7+ASf4RN89hGDkYyUOKZUcTFxTCMZMXFMqWJI1Z4gcTxKI+O4xsh//mkkI49r
fPvl8bSsnVPF9DuMf3+9ploxXmb//nlnK0lSM5utJEnNbLaSJDWz2UqS1Gw9Xvgfnh/PebiCdt1+
hcDKPYxyXGYKAuSlC4RL9jCScHcPASkYHTmvHOSapto/6F+vtYAUBcQ2+JwuEJ46Q3jqBJ/vCc4d
4VxExBFCSQcIMz3DOapx0AmuWQ461UJQ44AUjGSEHykcyVjcWcvhqdpIRt5ZCwEpqEVwCArHNcJI
xurYxa1hXCOFoOq7bn9O/a3PfXze2UqS1MxmK0lSM5utJEnNbLaSJDWz2UqS1Gw9QHIwglOQ3yB1
+RkSoHcwyhHTyOACKc87SMDuYFn5skIaeRmkkavjGiGNvG2URs5rF0ojw2d4gs/wDNc8QY3SxhER
B0oVQzr4QMlheD7PF6rl7w2mkfFc/vnSzxqde6nntY6RjMfi2MVjOXGcp4ZpJCONY4wYLIGHGo1W
rI9yrC6B71gQP/j99Zqzb37uY/DOVpKkZjZbSZKa2WwlSWpms5Ukqdl6gPFpERFPELx5hNp3CKzs
YdYbBaTo3+TPEEo5wXPZrXnwYoEQ1Lzw+zZPtaAXjWvcrhCQgnDNZcvfGwqWnSk8RSEoCB0d4VxE
xAHqBwoz0Tl4/YdiCIr3x+a1A+wrxpGLgzXPVKexixiCgt8NHJ6iEBTtrIVzxZ21NI7xpU4jGSHo
BLUr1cpBp46xixCeekVYCU+2ZKDgohjkui3vbCVJamazlSSpmc1WkqRmNltJkpoNA1K0C/M7BDpo
utRuzgNLM+yB3SA8dIIQzP05D0nslzwgsVJAajDpqrrPluAEKXhvKCBFwTKqHSmQRgEpOBfBgaUj
vA667gHeNwpBHeD7zdeEc9W9s4OAFIagYGc1B6RgLy2GoGq7Z080JYqCTMWdtKN6dRLUVgxB0e7Z
ao133b7md9Tbh7Lqj/cxeGcrSVIzm60kSc1stpIkNbPZSpLUzGYrSVKz9TBI6x0ueXL46Qxp5CXv
47s5T6tSGplGGZ4hHUojAPeQOKY08jJIG1MamTaT0lUpcUyjHC/w3tD7RrVTMTV8gmuO6tXE8ZG+
G8XEMaWDaSQj7palNPJg+ucJEse0s5r20mKNRivSrtvi2MXzRInjWi0iYrvCzlqs0UjG4ijHhr20
5Z21v8CYw9eMiPwIvLOVJKmZzVaSpGY2W0mSmtlsJUlqth4hBBAR8XzJ/9H+EQIrdxCe2k15j58i
DzPxuMb8mndLfs3dDAEpGMk4CkhR0ItgnAEDUjCuEcJTZwjznPFcMVg1CEgd4flQ0IlrdM3aaMVq
mOkIASlY84yjE0d1DDrBuNYTjlak3bN5KIlq1RAUj3Lk32/1kYzFoFNxtGK11jdWsWMk46+bd7aS
JDWz2UqS1MxmK0lSM5utJEnNbLaSJDVbD9MR/4Pn7T6vXfJe/Qhp5BXSyDTLkNPItRTrbs4T1ZQ4
Ho5rpERecV7jFQ5yGrl27ozjMeGzgBQzfU4v9dpZShVzyreWOKYaPx6dqy1yfzlbHbuYp3VxJCOk
gzFxDI93Li6Bp6TyBuMRX+qwPL6aKi6PZLxt4pi/UaMq/qIqX/e96HiW3tlKktTMZitJUjObrSRJ
zWy2kiQ1W4+jgBQECJ5hXOMeg0f0iHn/x4AU1Gif5w5GMq4QghqNY8S/YugsvA7edUu1WngKA1LF
8BSdi+AQFAWP6PPHcw0hKAo6nenxaA/saFwjhHJotCKHoGBcI4agaCRj/vumupeWQ071cY3lvbQd
Yxc/1M7ajxGCuvXz9M5WkqRmNltJkprZbCVJamazlSSpmc1WkqRm63E64H9wgDTf02WX1nZz3sdn
nFeYqy6P38+URs6vSanp8bhGqEGRAoLVdc2cR

In [217]:
# S-wave
function projected_dalitz(s,par)
    sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s*
        quadgk(zs->amplsp_analytic(s,zs,par),
                -0.999,0.999)[1]
end

plot(hists)
norm1 = quadgk(s->projected_dalitz(s,[[1,0,0]]),((mπ+mK)^2+0.01),((mB-mJpsi)^2-0.001))[1]
plot!(s->projected_dalitz(s,[[1,0,0]])/norm1, 
    ((mπ+mK)^2+0.01):0.02:((mB-mJpsi)^2-0.001),
    lab="j=1, l=0")
# D-wave
norm2 = quadgk(s->projected_dalitz(s,[[0,0,0.01]]),((mπ+mK)^2+0.01),((mB-mJpsi)^2-0.001))[1]
plot!(s->projected_dalitz(s,[[0,0,0.01]])/norm2,
    ((mπ+mK)^2+0.01):0.02:((mB-mJpsi)^2-0.001),
    lab="j=1, l=2")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.00 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
<polyline clip-path="url(#clip4102)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 58.7198,374.898 63.4499,371.247 68.1801,366.382 72.9102,360.813 77.6403,354.808 82.3704,348.537 87.1005,342.117 91.8306,335.631 96.5607,329.139 101.291,322.687 
 106.021,316.305 110.751,310.021 115.481,303.85 120.211,297.807 124.941,291.901 129.672,286.137 134.402,280.522 139.132,275.056 143.862,269.741 148.592,264.577 
 153.322,259.564 158.052,254.701 162.782,249.986 167.512,245.416 172.243,240.991 176.973,236.706 181.703,232.561 186.433,228.552 191.163,224.678 195.893,220.935 
 200.623,217.321 205.353,213.834 210.083,210.471 214.814,207.231 219.544,204.11 224.274,201.108 229.004,198.222 233.734,195.45 238.464,192.79 243.194,190.241 
 247.924,187.801 252.654,185.47 257.385,183.244 262.115,181.124 266.845,179.108 271.575,177.195 276.305,175.383 281.035,173.673 285.765,172.064 290.495,170.554 
 295.225,169.143 299.956,167.831 304.686,166.618 309.416,165.502 314.146,164.484 318.876,163.565 323.606,162.742 328.336,162.018 333.066,161.392 337.796,160.865 
 342.527,160.436 347.257,160.107 351.987,159.877 356.717,159.749 361.447,159.722 366.177,159.798 370.907,159.977 375.637,160.262 380.367,160.653 385.098,161.152 
 389.828,161.76 394.558,162.481 399.288,163.315 404.018,164.266 408.748,165.335 413.478,166.525 418.208,167.84 422.938,169.283 427.669,170.858 432.399,172.568 
 437.129,174.418 441.859,176.412 446.589,178.558 451.319,180.859 456.049,183.323 460.779,185.958 465.509,188.771 470.24,191.771 474.97,194.97 479.7,198.379 
 484.43,202.011 489.16,205.883 493.89,210.011 498.62,214.417 503.35,219.125 508.08,224.165 512.811,229.573 517.541,235.393 522.271,241.678 527.001,248.501 
 531.731,255.954 536.461,264.166 541.191,273.319 545.921,283.698 550.651,295.779 555.382,310.504 560.112,330.378 
 "/>
<polyline clip-path="url(#clip4102)" style="stroke:#c271d2; stroke-width:1; stroke-opacity:1; fill:none" points="
 58.7198,371.316 63.4499,353.959 68.1801,331.849 72.9102,307.713 77.6403,282.954 82.3704,258.414 87.1005,234.631 91.8306,211.947 96.5607,190.582 101.291,170.664 
 106.021,152.264 110.751,135.407 115.481,120.089 120.211,106.284 124.941,93.9506 129.672,83.0364 134.402,73.4827 139.132,65.2267 143.862,58.2029 148.592,52.345 
 153.322,47.5868 158.052,43.8631 162.782,41.11 167.512,39.2657 172.243,38.2703 176.973,38.0661 181.703,38.5981 186.433,39.8134 191.163,41.6617 195.893,44.0951 
 200.623,47.0679 205.353,50.537 210.083,54.4612 214.814,58.8015 219.544,63.5212 224.274,68.5853 229.004,73.9607 233.734,79.6162 238.464,85.5223 243.194,91.6511 
 247.924,97.9762 252.654,104.473 257.385,111.117 262.115,117.887 266.845,124.762 271.575,131.722 276.305,138.749 281.035,145.824 285.765,152.932 290.495,160.057 
 295.225,167.183 299.956,174.298 304.686,181.389 309.416,188.443 314.146,195.449 318.876,202.396 323.606,209.274 328.336,216.074 333.066,222.787 337.796,229.

## Example of the interest
Consider two resonances:
 * $K^*(1410)$, $m = 1.414\,$MeV, $\Gamma = 232\,$MeV
 * $K_2^*(1430)$, $m = 1.432\,$MeV, $\Gamma = 109\,$MeV

In [219]:
mKs = 1.414; ΓKs = 0.232;
mKs2 = 1.432; ΓKs2 = 0.109;
function bwDalitz(s,msq,Γ)
    phsp = sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s
    phsp*abs(bw(s,msq,Γ))^2
end
plot(s->bwDalitz(s,mKs^2,ΓKs), (mπ+mK)^2:0.01:(mB-mJpsi)^2, lab="K*(1410)")
plot!(s->bwDalitz(s,mKs2^2,ΓKs2), (mπ+mK)^2:0.01:(mB-mJpsi)^2, lab="K2*(1430)", legend=:topleft)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0 
 
 
 50 
 
 
 100 
 
 
 150 
 
 
 200 
 
<polyline clip-path="url(#clip4502)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.5169,375.813 38.0627,374.752 40.6085,374.309 43.1543,373.966 45.7001,373.675 48.2458,373.416 50.7916,373.18 53.3374,372.961 55.8832,372.755 58.429,372.559 
 60.9748,372.372 63.5206,372.192 66.0663,372.018 68.6121,371.849 71.1579,371.685 73.7037,371.524 76.2495,371.366 78.7953,371.21 81.3411,371.057 83.8868,370.905 
 86.4326,370.755 88.9784,370.607 91.5242,370.459 94.07,370.312 96.6158,370.166 99.1616,370.02 101.707,369.874 104.253,369.728 106.799,369.582 109.345,369.436 
 111.89,369.289 114.436,369.142 116.982,368.994 119.528,368.845 122.074,368.695 124.619,368.544 127.165,368.392 129.711,368.238 132.257,368.083 134.803,367.927 
 137.348,367.768 139.894,367.608 142.44,367.447 144.986,367.283 147.531,367.117 150.077,366.948 152.623,366.778 155.169,366.605 157.715,366.429 160.26,366.251 
 162.806,366.07 165.352,365.886 167.898,365.699 170.444,365.509 172.989,365.316 175.535,365.119 178.081,364.918 180.627,364.714 183.172,364.506 185.718,364.294 
 188.264,364.078 190.81,363.858 193.356,363.633 195.901,363.404 198.447,363.169 200.993,362.93 203.539,362.686 206.085,362.436 208.63,362.181 211.176,361.92 
 213.722,361.653 216.268,361.38 218.813,361.101 221.359,360.815 223.905,360.523 226.451,360.223 228.997,359.916 231.542,359.601 234.088,359.279 236.634,358.948 
 239.18,358.609 241.726,358.262 244.271,357.905 246.817,357.539 249.363,357.163 251.909,356.777 254.454,356.38 257,355.973 259.546,355.554 262.092,355.124 
 264.638,354.682 267.183,354.227 269.729,353.76 272.275,353.279 274.821,352.783 277.367,352.274 279.912,351.749 282.458,351.209 285.004,350.653 287.55,350.08 
 290.095,349.49 292.641,348.882 295.187,348.256 297.733,347.61 300.279,346.944 302.824,346.258 305.37,345.55 307.916,344.821 310.462,344.068 313.008,343.292 
 315.553,342.492 318.099,341.667 320.645,340.816 323.191,339.938 325.736,339.033 328.282,338.1 330.828,337.138 333.374,336.147 335.92,335.125 338.465,334.072 
 341.011,332.988 343.557,331.873 346.103,330.724 348.649,329.544 351.194,328.33 353.74,327.084 356.286,325.805 358.832,324.493 361.377,323.15 363.923,321.776 
 366.469,320.372 369.015,318.939 371.561,317.48 374.106,315.996 376.652,314.49 379.198,312.965 381.744,311.425 384.289,309.874 386.835,308.317 389.381,306.759 
 391.927,305.207 394.473,303.667 397.018,302.146 399.564,300.654 402.11,299.199 404.656,297.79 407.202,296.437 409.747,295.152 412.293,293.944 414.839,292.826 
 417.385,291.809 419.93,290.903 422.476,290.12 425.022,289.47 427.568,288.962 430.114,288.606 432.659,288.408 435.205,288.375 437.751,288.511 440.297,288.819 
 442.843,289.3 445.388,289.952 447.934,290.773 450.48,291.757 453.026,292.899 455.571,294.19 458.117,295.62 460.663,297.179 463.209,298.854 465.755,300.633 
 468.3,302.504 470.846,304.452 473.392,306.465 475.938,308.53 478.484,310.635 481.029,312.766 483.575,314.914 486.121,317.068 488.667,319.218 491.212,321.356 
 493.758,323.474 496.304,325.565 498.85,327.625 501.396,329.647 503.941,331.628 506.487,333.564 509.033,335.453 511.579,337.294 514.125,339.083 516.67,340.822 
 519.216,342.509 521.762,344.145 524.308,345.73 526.853,347.265 529.399,348.751 531.945,350.188 534.491,351.58 537.037,352.928 539.582,354.233 542.128,355.497 
 544.674,356.724 547.22,357.915 549.766,359.073 552.311,360.202 554.857,361.304 557.403,362.384 559.949,363.445 562.494,364.494 565.04,365.537 567.586,366.582 
 570.132,367.643 572.678,368.737 575.223,369.896 577.769,371.184 580.315,372.781 
 "/>
<polyline clip-path="url(#clip4502)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.5169,375.813 38.0627,374.784 40.6085,374.354 43.1543,374.022 45.7001,373.739 48.2458,373.488 50.7916,373.259 53.3374,3

In [101]:
plot(0.5:0.1:5, -.99:0.03:0.99,
    (s,zs)->(sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s*
amplsp_analytic(s,zs,[[0,0,1]])), t=:heatmap)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 1 
 
 
 2 
 
 
 3 
 
 
 4 
 
 
 5 
 
 
 -1.0 
 
 
 -0.5 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
<image width="475" height="364" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAdsAAAFsCAYAAACEtRP5AAAgAElEQVR4nO3du26k6XGH8foOzePM
rGSsYK0cObCdOVNiCPBN+Aqc+0J9CQ4UGIaFXWlPMySb3f2dHLQcGILWT2nf2h5ynl9ceF92s8ni
h/lPVfcvn/3bFtC//t3vUd0//ebfUd2bf36kV8fxH3+D6tZf/hrV3dz8Db6760dUt8zs9RwPX/K7
3/8W1Y3f/ier++q/+N1ffoXqlq86VHf6/c/w3fuvWe2Hb2Hdw1t893ePb1Ddt4c7VPf98Zrffdqx
M08Dqns/se9NRMTDxOoe55WdNy/47sc4orqnbo/q9t0TvvsQD6juuLG75/UZ3z2vB1a3sfdnXVld
RMS6sW/4trHv4xb8+x34TPZZi8BtLGJjtVvmTKBvepokSfoTNltJkorZbCVJKmazlSSpmM1WkqRi
42nliaun6QrVPX/Pkp/3X/JU7u7z/0B1E0yaHd78At+99Swh2k0sATnsv8V3D9+zRPDwDXsvuz98
g+9e/sDqpm/Z9/sAPxcREfsP96jueX+L6p4ON/hu/DmfWUr9eWHJ4YiIw8L+/j2uMAFOw5wRMcHf
BfR3xpRIp86wdo4Z1S2w7lwLk7EwQbviBG0ibbuxutYJ2v89tW3dp8knW0mSitlsJUkqZrOVJKmY
zVaSpGLjaeX/mL+fWVBo/4GNu5u/ZuGWiIjd737H6mBAan3DRk9GRGw9C7h0ExuV1j+9x3f3779n
hd99QGXLN/zvKxp8On7H6p7h5yIiYv/EAlI0+PR04iMT980DUvw9Pyws+HSE2aNTYoIeDVNNMKxD
Q0/nWhh86mhIiQek+DhCGFKC70+mlgefeEipJkx1IfD3/iX5ZCtJUjGbrSRJxWy2kiQVs9lKklTM
ZitJUrGRJgsjIp5g+vLpiS3VPn79Gb57eMvGEfbrf7O6++/w3dvA/ibpJpiA3PPl0tsDTGm+Z9+b
+XueCKYp4z0cw/j0wBLGERF7mjI+0jQySxhHROzh53wPxzCWpJHhj21mHCv9XdC6LiJi7ukYRrbw
PDOucYWp6bX5wnOeCOZJ6Ewqt3HCOXH3ZZPQl7nbJ1tJkorZbCVJKmazlSSpmM1WkqRi4ykRJKDj
Gh+fWUBq/907fPfu7R7VjesDquvv+MjEgOtIN5bdiPXAQjAREesTCwDNMHx0hHtiI/h4xf0jrHvm
4zkfD6z2EY5hfIKf3YhE8Glmf6vS0FNExAFma+i4ximxz7b1ntq5479bWu+pTQWkLjmuEe+zLdgp
S3NP/MQCH/8YRsonW0mSitlsJUkqZrOVJKmYzVaSpGI2W0mSio2nxILnp5klYx9gkvTpPRvzFxFx
9fUB1d3A0XjDLVv0HhERPRypBu9eD3x04Lxn7/npib3nh8fEyMRHmCqHKWOaMI6IeDyylPEjXPRO
RzCea9uOYXxOpJHxUniYHD4lkrETTOXyNDKM50fEApPLdFwjTRhHRGzw9fBF74kI+AWXxzcf16gf
5JOtJEnFbLaSJBWz2UqSVMxmK0lSsXGCgYOIiD0MejzQcMsTD+tcf8cCTTSkNN6c8N1dz0IMKxzz
t5z46MDpmb2Xx2cWpHre85DSM9wpS4NPD3D3bETEw8ReN949C0NPEe2DTzT0FBFxaryn9rTysA4N
S9LRilNiZOLceE/tVjKuEX4jU+MaLxlSan2mQaof4pOtJEnFbLaSJBWz2UqSVMxmK0lSMZutJEnF
xlNipNp+hmlkOELvw56NA4yIuNrBtCJMku6u+evuOjhSbWV3zxNPI5+O7L08wiXqzzApHhGxp2lk
ePfDiY+ppCnjJ5gyfoKfi/Pd7HP+DMOph0waeaEp44JxjTBtO8HRipnl8XgMI0wZ58Y1tl0KnxnX
2P7ul5Barjrz4+aTrSRJxWy2kiQVs9lKklTMZitJUrHUuEYaCHmE4ZYPRz468HoPgxEwpLQ78PAG
DUjRu2c41jEiYoJhqgPd65oJKcHg0xO8+ymxU5Z+hmhA6nnmf1fSMYw0+HRKBKSOjccw0tDTuRYG
n/BoxcQ+28bBpzXxujc8rrFtmOmPxbSwcV22Vj+WT7aSJBWz2UqSVMxmK0lSMZutJEnFxrnje11p
IORxogEpvt/0aoC7NmFI6WrgAam+Y0GCdWPBmkxA6jizgBSt28O6iIhnGM7aw9eTCUjR4NMeToai
u5gj+Oec7qmloacIHnyiu2dPiUlKU88CTXSC1CUDUpmQ0voCJkjxA/lnDVcmztSf55OtJEnFbLaS
JBWz2UqSVMxmK0lSMZutJEnFxikyaWSWSnuEo/EeEntdR7hblaeReUqzg2nkjaaR4dcYEXGCSd8D
TPoeFp4IPsC79zQ5DOsi+P5ZvHuWh895GhmmjI8wYRwRcYRp2xNM706JfdUzPHOBvzPoWMeIiHWD
aWSaWk6Na2ydMq4Ymfjx75St2aX7evhkK0lSMZutJEnFbLaSJBWz2UqSVGycUwEpFhB4hEGYD3Cs
Y0TE2LOdqcsG99n2PEDRfFxjIiA1wdrTCoNUiVGRBxhSesZhpvY7ZemOZRp6iog4wiAgDT7REYwR
fAwj3T2bCUjRWr7PlqfScEAKj2tMBKTge46DVIlRkRschcjDR68spPSKXo5PtpIkFbPZSpJUzGYr
SVIxm60kScVstpIkFRuXjaeRn2HCjy7+fkiM7xt6NtqRJn13PU8Mth7XuMC6iIhpZbUn+Lppwjgi
4oDPZF8jrYvg4xVpyviQWOBOa2ka+ZhI5dJl7yeYHJ5SC9zZ7wK6FJ4mjCP4eEWcHM4scG88rjE3
thDW4gXur21U5OuJI/tkK0lSMZutJEnFbLaSJBWz2UqSVGxcNh6gOMCgx35moxUfJh6YGToWppph
oGjX83947+E/0tMTLxmQOsHzIiKONPgEsyjHxMjE5jtl4QjG85kw+IR3z/IXfupYSAmPVoTnRUTQ
0a0lASk8rpGOTMyMa2y9z5aHs1oHhVJxoubZo5cRZrrUV+mTrSRJxWy2kiQVs9lKklTMZitJUjGb
rSRJxUaaLIyIOMK04vNyg+oeE8vE+47V0vRuLo3M8DQyvjpmmFyeYAAyk0Y+4URw27rz3Y1HJibG
NR5hMpaOYaQ/NxERE04jszqaMD7XwqXwNDmcGFPJxzC2XfR+Loa1dNE7Hq1YsRS+YlyjWvDJVpKk
YjZbSZKK2WwlSSpms5UkqdiYGal26I+o7nlhgYN9JiAF6+i4xrHnQSFaSnMRmWFu8K2MCd5Ng1QR
ESd6Nw5n8UDGCQef2o5WjIg4wWDPCQaKaJgpgu+ppcGnTECKj0ykdfw9p7V0DCMNUp1rW49rfF0h
pdx+3kv5+L9Gn2wlSSpms5UkqZjNVpKkYjZbSZKKjWtin+2xO6C6/crOvFl4r+/g3wU0KLTj+ajo
ErVEIieEp03NMLsxJ+6e4Bc6wWQYDT2da+GZOMyUCUixzy/dPUtDTxERc7AQIg9I8bvxnlr4XtK6
iERICe+z5Z81fmbBFKfmZ76ucNZr4pOtJEnFbLaSJBWz2UqSVMxmK0lSMZutJEnFUuMaT8HSyM8w
1fg0j/huiu5/HRIRYzyuEZ6XWHcZKyymqWWaHI6ImBunkafUyERWS8/MJIJpyniCZ04wYRzRPmWc
+fluPa5xS9zdegwjH62YGUdIxzq2T/niM

## Compare three amplitudes
* Our amplitude, Eq. (B1)
* LS-amplitude, Eq. (13)
* Tensor amplitude, Eq. (19)

Partial wave expansion for the amplitude
$$
A_\lambda(s,\cos\theta) = \frac{1}{4\pi}\sum (2j+1) A_\lambda^j(s) d_{\lambda 0}^j(\theta)
$$

Integrating over muons is equivalent to summing over helicity
$$
\frac{\textrm{d}I}{\textrm{d}s} = N\big( |A_0^1(s)|^2 + 2|A_1(s)|^2 \big) \frac{\lambda_{\pi K}^{1/2}(s)
\lambda_{\psi B}^{1/2}(s)}{s}
$$
Below, I demonstrate it numerically.

In [238]:
# our amplitude
function A10_our(s,cS,cD)
    j = 1
    lam=0
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    K00 = 2*mJpsi/sqrt(λ(s,mJpsi^2,mB^2))
    K00*(p*q)^j*(s+mJpsi^2-mB^2)/(2*mJpsi^2)*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,0,j,0)*cS+
        λ(s,mJpsi^2,mB^2)*sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,0,j,0)*cD)
end
function A11_our(s,cS,cD)
    j = 1
    lam=0
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    K10 = q
    K10*(p*q)^(j-1)*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,1,j,1)*cS+
        λ(s,mJpsi^2,mB^2)*sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,1,j,1)*cD)
end

Msq_our(s,zs,cS,cD) = 3*(abs(A10_our(s,cS,cD)*Wignerd(1,0,0,zs))^2+2*abs(A11_our(s,cS,cD)*Wignerd(1,1,0,zs))^2)

function project_general(s,cS,cD,A10,A11)
    ph(s)*(abs(A10(s,cS,cD))^2+2*abs(A11(s,cS,cD))^2)
end

project_general (generic function with 1 method)

In [241]:
s_th = (mπ+mK)^2+0.0001
s_en = (mB-mJpsi)^2-0.0001;

# first, our amplitude from tensor form
ph(s) = sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s;
function project(s)
    ph(s)*quadgk(zs->Msq_our(s,zs,0,1),-0.999,0.999)[1]
end
norm1 = quadgk(s->project(s),s_th,s_en)[1]
plot(s->project(s)/norm1,s_th:0.03:s_en, lab="sum of helicities", title="JPAC model")

# second, numerical projection, helicity
function project(s)
    ph(s)*quadgk(zs->amplsp_analytic(s,zs,[[0,0,1]]),-0.999,0.999)[1]
end
norm1 = quadgk(s->project(s),s_th,s_en)[1]
plot!(s->project(s)/norm1,s_th:0.03:s_en, lab="full model with tensors")

# analytic integrals
norm1 = quadgk(s->project_general(s,0,1,A10_our,A11_our),s_th,s_en)[1]
plot!(s->project_general(s,0,1,A10_our,A11_our)/norm1,s_th:0.03:s_en, lab="analyt. cosT integr")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 JPAC model 
 
<polyline clip-path="url(#clip7102)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,353.436 47.6613,318.727 55.381,281.032 63.1006,243.778 70.8202,208.663 78.5398,176.552 86.2594,147.845 93.979,122.671 101.699,100.998 
 109.418,82.6955 117.138,67.5813 124.858,55.442 132.577,46.0521 140.297,39.1841 148.016,34.6147 155.736,32.1295 163.456,31.5244 171.175,32.6076 178.895,35.1997 
 186.614,39.1338 194.334,44.255 202.054,50.4201 209.773,57.497 217.493,65.3638 225.213,73.9085 232.932,83.028 240.652,92.6277 248.371,102.621 256.091,112.927 
 263.811,123.474 271.53,134.194 279.25,145.026 286.969,155.915 294.689,166.809 302.409,177.661 310.128,188.429 317.848,199.074 325.568,209.562 333.287,219.859 
 341.007,229.939 348.726,239.774 356.446,249.342 364.166,258.622 371.885,267.596 379.605,276.248 387.325,284.564 395.044,292.532 402.764,300.141 410.483,307.384 
 418.203,314.253 425.923,320.743 433.642,326.851 441.362,332.575 449.081,337.914 456.801,342.867 464.521,347.439 472.24,351.63 479.96,355.448 487.68,358.896 
 495.399,361.983 503.119,364.718 510.838,367.11 518.558,369.171 526.278,370.915 533.997,372.356 541.717,373.511 549.436,374.401 557.156,375.046 564.876,375.474 
 572.595,375.715 580.315,375.809 
 "/>
<polyline clip-path="url(#clip7102)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,353.436 47.6613,318.727 55.381,281.032 63.1006,243.778 70.8202,208.663 78.5398,176.552 86.2594,147.845 93.979,122.671 101.699,100.998 
 109.418,82.6955 117.138,67.5813 124.858,55.442 132.577,46.0521 140.297,39.1841 148.016,34.6147 155.736,32.1295 163.456,31.5244 171.175,32.6076 178.895,35.1997 
 186.614,39.1338 194.334,44.255 202.054,50.4201 209.773,57.497 217.493,65.3638 225.213,73.9085 232.932,83.028 240.652,92.6277 248.371,102.621 256.091,112.927 
 263.811,123.474 271.53,134.194 279.25,145.026 286.969,155.915 294.689,166.809 302.409,177.661 310.128,188.429 317.848,199.074 325.568,209.562 333.287,219.859 
 341.007,229.939 348.726,239.774 356.446,249.342 364.166,258.622 371.885,267.596 379.605,276.248 387.325,284.564 395.044,292.532 402.764,300.141 410.483,307.384 
 418.203,314.253 425.923,320.743 433.642,326.851 441.362,332.575 449.081,337.914 456.801,342.867 464.521,347.439 472.24,351.63 479.96,355.448 487.68,358.896 
 495.399,361.983 503.119,364.718 510.838,367.11 518.558,369.171 526.278,370.915 533.997,372.356 541.717,373.511 549.436,374.401 557.156,375.046 564.876,375.474 
 572.595,375.715 580.315,375.809 
 "/>
<polyline clip-path="url(#clip7102)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,353.43 47.6613,318.712 55.381,281.009 63.1006,243.746 70.8202,208.625 78.5398,176.509 86.2594,147.799 93.979,122.623 101.699,100.949 
 109.418,82.6468 117.138,67.5338 124.858,55.3964 132.577,46.0092 140.297,39.1443 148.016,34.5785 155.736,32.0971 163.456,31.4961 171.175,32.5835 178.895,35.1799 
 186.614,39.1182 194.334,44.2437 202.054,50.4129 209.773,57.4937 217.493,65.3644 225.213,73.9127 232.932,83.0356 240.652,92.6383 248.371,102.634 256.091,112.943 
 263.811,123.492 271.53,134.215 279.25,145.049 286.969,155.939 294.689,166.834 302.409,177.687 310.128,188.456 317.848,199.102 325.568,209.59 333.287,219.887 
 341.007,229.967 348.726,239.802 356.446,249.369 364.166,258.649 371.885,267.622 379.605,276.273 387.325,284.588 395.044,292.555 402.764,300.163 410.483,307.404 
 418.203,314.272 425.923,320.761 433.642,326.868 441.362,332.59 449.081,337.927 456.801,342.879 464.521,347.449 472.24,351.64 479.96,355.456 487.68,358.903 
 495.399,361.989 503.119,364.723 510.838,367.114 518.558,369.174 526.278,370.917 533.997,372.357 541.717,373.512 549.436,374.401 557.156,375.

In [155]:
# LS amplitude
function A1λ_LS(s,cS,cD,lam)
    j = 1
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    p^(j-1)*q^j*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,lam,j,lam)*cS+
        sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,lam,j,lam)*p^2*cD)
end
A10_LS(s,cS,cD) = A1λ_LS(s,cS,cD,0)
A11_LS(s,cS,cD) = A1λ_LS(s,cS,cD,1)

# tensor amplitude
function A10_ten(s,cS,cD)
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    γ = (mB^2+s-mJpsi^2)/(2*mB*sqrt(s))
    E1 = (s+mJpsi^2-mB^2)/(2*sqrt(s))
    q*(E1/mJpsi*cS - γ*p^2*s/mB^2*(s-mJpsi^2-mB^2)/(2*mJpsi*mB)*cD)
end
function A11_ten(s,cS,cD)
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    q*(cS + p^2/2*s/mB^2*cD)
end

A11_ten (generic function with 1 method)

In [158]:
plot([
    s->project_general(s,0,1,A10_our,A11_our),
    s->project_general(s,0,1,A10_LS,A11_LS),
    s->project_general(s,0,1,A10_ten,A11_ten)
    ],
2.4:0.00003:((mB-mJpsi)^2),lab=["new","LS","tensor"])

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 2.40 
 
 
 2.42 
 
 
 2.44 
 
 
 2.46 
 
 
 2.48 
 
 
 2.50 
 
 
 2.52 
 
 
 30 
 
 
 60 
 
 
 90 
 
 
 120 
 
 
 150 
 
<polyline clip-path="url(#clip7102)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.5169,11.811 35.6357,12.0078 35.7544,12.2045 35.8732,12.4012 35.9919,12.5978 36.1106,12.7943 36.2294,12.9908 36.3481,13.1872 36.4669,13.3836 36.5856,13.5799 
 36.7044,13.7761 36.8231,13.9722 36.9418,14.1683 37.0606,14.3644 37.1793,14.5603 37.2981,14.7562 37.4168,14.9521 37.5356,15.1479 37.6543,15.3436 37.7731,15.5392 
 37.8918,15.7348 38.0105,15.9304 38.1293,16.1258 38.248,16.3212 38.3668,16.5166 38.4855,16.7118 38.6043,16.9071 38.723,17.1022 38.8418,17.2973 38.9605,17.4923 
 39.0792,17.6873 39.198,17.8822 39.3167,18.077 39.4355,18.2718 39.5542,18.4665 39.673,18.6611 39.7917,18.8557 39.9105,19.0502 40.0292,19.2447 40.1479,19.4391 
 40.2667,19.6334 40.3854,19.8277 40.5042,20.0219 40.6229,20.216 40.7417,20.4101 40.8604,20.6041 40.9792,20.7981 41.0979,20.992 41.2166,21.1858 41.3354,21.3796 
 41.4541,21.5733 41.5729,21.7669 41.6916,21.9605 41.8104,22.154 41.9291,22.3475 42.0478,22.5409 42.1666,22.7342 42.2853,22.9274 42.4041,23.1206 42.5228,23.3138 
 42.6416,23.5069 42.7603,23.6999 42.8791,23.8928 42.9978,24.0857 43.1165,24.2785 43.2353,24.4713 43.354,24.664 43.4728,24.8566 43.5915,25.0492 43.7103,25.2417 
 43.829,25.4341 43.9478,25.6265 44.0665,25.8189 44.1852,26.0111 44.304,26.2033 44.4227,26.3954 44.5415,26.5875 44.6602,26.7795 44.779,26.9715 44.8977,27.1634 
 45.0165,27.3552 45.1352,27.5469 45.2539,27.7386 45.3727,27.9303 45.4914,28.1218 45.6102,28.3133 45.7289,28.5048 45.8477,28.6962 45.9664,28.8875 46.0851,29.0787 
 46.2039,29.2699 46.3226,29.4611 46.4414,29.6521 46.5601,29.8431 46.6789,30.0341 46.7976,30.225 46.9164,30.4158 47.0351,30.6065 47.1538,30.7972 47.2726,30.9879 
 47.3913,31.1784 47.5101,31.3689 47.6288,31.5594 47.7476,31.7498 47.8663,31.9401 47.9851,32.1303 48.1038,32.3205 48.2225,32.5107 48.3413,32.7007 48.46,32.8907 
 48.5788,33.0807 48.6975,33.2706 48.8163,33.4604 48.935,33.6501 49.0538,33.8398 49.1725,34.0294 49.2912,34.219 49.41,34.4085 49.5287,34.598 49.6475,34.7873 
 49.7662,34.9767 49.885,35.1659 50.0037,35.3551 50.1224,35.5443 50.2412,35.7333 50.3599,35.9223 50.4787,36.1113 50.5974,36.3002 50.7162,36.489 50.8349,36.6777 
 50.9537,36.8664 51.0724,37.0551 51.1911,37.2436 51.3099,37.4321 51.4286,37.6206 51.5474,37.809 51.6661,37.9973 51.7849,38.1855 51.9036,38.3737 52.0224,38.5619 
 52.1411,38.7499 52.2598,38.938 52.3786,39.1259 52.4973,39.3138 52.6161,39.5016 52.7348,39.6894 52.8536,39.8771 52.9723,40.0647 53.0911,40.2523 53.2098,40.4398 
 53.3285,40.6272 53.4473,40.8146 53.566,41.002 53.6848,41.1892 53.8035,41.3764 53.9223,41.5636 54.041,41.7506 54.1597,41.9376 54.2785,42.1246 54.3972,42.3115 
 54.516,42.4983 54.6347,42.6851 54.7535,42.8718 54.8722,43.0584 54.991,43.245 55.1097,43.4315 55.2284,43.618 55.3472,43.8044 55.4659,43.9907 55.5847,44.177 
 55.7034,44.3632 55.8222,44.5493 55.9409,44.7354 56.0597,44.9214 56.1784,45.1074 56.2971,45.2933 56.4159,45.4791 56.5346,45.6649 56.6534,45.8506 56.7721,46.0362 
 56.8909,46.2218 57.0096,46.4073 57.1284,46.5928 57.2471,46.7782 57.3658,46.9635 57.4846,47.1488 57.6033,47.334 57.7221,47.5192 57.8408,47.7043 57.9596,47.8893 
 58.0783,48.0743 58.197,48.2592 58.3158,48.444 58.4345,48.6288 58.5533,48.8135 58.672,48.9982 58.7908,49.1828 58.9095,49.3673 59.0283,49.5518 59.147,49.7362 
 59.2657,49.9205 59.3845,50.1048 59.5032,50.289 59.622,50.4732 59.7407,50.6573 59.8595,50.8413 59.9782,51.0253 60.097,51.2092 60.2157,51.3931 60.3344,51.5769 
 60.4532,51.7606 60.5719,51.9442 60.6907,52.1279 60.8094,52.3114 60.9282,52.4949 61.0469,52.6783 61.1657,52.8616 61.2844,53.0449 61.4031,53.2282 61.5219,53.4113 
 61.6406,53.5945 61.7594,53.7775 61.8781,53.9605 61.9969,54.1434 62.1156,54.3263 62.2343,54.5091 62.3531,54.6918 

In [245]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our),
    s->project_general(s,0,1,A10_LS,A11_LS),
    s->project_general(s,0,1,A10_ten,A11_ten)
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.03:s_en, lab=["JPAC","LS","tensor"])

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
<polyline clip-path="url(#clip7702)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 38.8117,375.812 46.4385,356.383 54.0653,326.247 61.6922,293.519 69.319,261.175 76.9458,230.689 84.5726,202.812 92.1994,177.89 99.8262,156.037 107.453,137.223 
 115.08,121.337 122.707,108.218 130.333,97.6828 137.96,89.5345 145.587,83.5756 153.214,79.6124 160.841,77.4585 168.467,76.9368 176.094,77.8807 183.721,80.1344 
 191.348,83.553 198.975,88.002 206.601,93.357 214.228,99.5033 221.855,106.335 229.482,113.755 237.109,121.674 244.735,130.01 252.362,138.686 259.989,147.635 
 267.616,156.792 275.243,166.099 282.87,175.503 290.496,184.956 298.123,194.413 305.75,203.834 313.377,213.182 321.004,222.423 328.63,231.526 336.257,240.465 
 343.884,249.214 351.511,257.751 359.138,266.056 366.764,274.111 374.391,281.9 382.018,289.409 389.645,296.627 397.272,303.542 404.898,310.146 412.525,316.431 
 420.152,322.393 427.779,328.026 435.406,333.326 443.032,338.293 450.659,342.926 458.286,347.225 465.913,351.192 473.54,354.829 481.166,358.141 488.793,361.134 
 496.42,363.813 504.047,366.185 511.674,368.261 519.301,370.049 526.927,371.562 534.554,372.812 542.181,373.815 549.808,374.586 557.435,375.147 565.061,375.518 
 572.688,375.727 580.315,375.809 
 "/>
<polyline clip-path="url(#clip7702)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 38.8117,375.806 46.4385,330.494 54.0653,267.916 61.6922,207.874 69.319,155.632 76.9458,112.567 84.5726,78.5187 92.1994,52.7035 99.8262,34.1194 107.453,21.7384 
 115.08,14.5927 122.707,11.811 130.333,12.6282 137.96,16.3831 145.587,22.5103 153.214,30.5293 160.841,40.0346 168.467,50.6845 176.094,62.1925 183.721,74.3191 
 191.348,86.8641 198.975,99.6615 206.601,112.573 214.228,125.485 221.855,138.304 229.482,150.954 237.109,163.371 244.735,175.507 252.362,187.322 259.989,198.784 
 267.616,209.87 275.243,220.562 282.87,230.848 290.496,240.72 298.123,250.173 305.75,259.206 313.377,267.819 321.004,276.016 328.63,283.803 336.257,291.185 
 343.884,298.169 351.511,304.766 359.138,310.983 366.764,316.83 374.391,322.319 382.018,327.46 389.645,332.264 397.272,336.743 404.898,340.907 412.525,344.768 
 420.152,348.338 427.779,351.628 435.406,354.649 443.032,357.414 450.659,359.933 458.286,362.217 465.913,364.278 473.54,366.127 481.166,367.775 488.793,369.232 
 496.42,370.509 504.047,371.618 511.674,372.568 519.301,373.37 526.927,374.036 534.554,374.576 542.181,375.001 549.808,375.322 557.435,375.551 565.061,375.699 
 572.688,375.782 580.315,375.813 
 "/>
<polyline clip-path="url(#clip7702)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 38.8117,375.813 46.4385,359.587 54.0653,334.055 61.6922,305.879 69.319,277.55 76.9458,250.356 84.5726,224.99 92.1994,201.817 99.8262,181.001 107.453,162.584 
 115.08,146.534 122.707,132.769 130.333,121.18 137.96,111.645 145.587,104.031 153.214,98.2037 160.841,94.0316 168.467,91.3852 176.094,90.14 183.721,90.1772 
 191.348,91.3838 198.975,93.6532 206.601,96.8845 214.228,100.983 221.855,105.859 229.482,111.43 237.109,117.616 244.735,124.343 252.362,131.544 259.989,139.152 
 267.616,147.107 275.243,155.352 282.87,163.834 290.496,172.502 298.123,181.31 305.75,190.213 313.377,199.17 321.004,208.142 328.63,217.095 336.257,225.993 
 343.884,234.805 351.511,243.502 359.138,252.056 366.764,260.443 374.391,268.637 382.018,276.619 389.645,284.367 397.272,291.863 404.898,299.09 412.525,306.034 
 420.152,312.68 427.779,319.016 435.406,325.032 443.032,330.719 450.659,336.068 458.286,341.073 465.913,345.731 473.54,350.036 481.166,353.989 488.793,357.588 
 496.42,360.835 504.047,363.734 511.674,366.289 519.301,368.506 526.927,370.396 534.554,371.97 542.181,373.241 549.808,374.226 557.435,374.946 565.061,375.427 
 572.688,375.699 

In [246]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2),
    s->project_general(s,0,1,A10_LS,A11_LS)*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2),
    s->project_general(s,0,1,A10_ten,A11_ten)*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2)
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.03:s_en, lab=["JPAC","LS","tensor"])

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.1 
 
 
 0.2 
 
 
 0.3 
 
 
 0.4 
 
 
 0.5 
 
 
 0.6 
 
 
 0.7 
 
<polyline clip-path="url(#clip7902)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,370.931 47.6613,362.718 55.381,352.962 63.1006,342.346 70.8202,331.249 78.5398,319.9 86.2594,308.445 93.979,296.981 101.699,285.576 
 109.418,274.276 117.138,263.113 124.858,252.113 132.577,241.294 140.297,230.668 148.016,220.248 155.736,210.042 163.456,200.057 171.175,190.299 178.895,180.774 
 186.614,171.487 194.334,162.442 202.054,153.645 209.773,145.098 217.493,136.808 225.213,128.777 232.932,121.012 240.652,113.516 248.371,106.294 256.091,99.3514 
 263.811,92.6937 271.53,86.3264 279.25,80.2554 286.969,74.4871 294.689,69.0282 302.409,63.8858 310.128,59.0674 317.848,54.5812 325.568,50.4359 333.287,46.6407 
 341.007,43.2056 348.726,40.1412 356.446,37.4592 364.166,35.1721 371.885,33.2932 379.605,31.8374 387.325,30.8206 395.044,30.2603 402.764,30.1757 410.483,30.5879 
 418.203,31.5203 425.923,32.9989 433.642,35.0527 441.362,37.7143 449.081,41.0205 456.801,45.0135 464.521,49.7415 472.24,55.2603 479.96,61.6355 487.68,68.9447 
 495.399,77.2812 503.119,86.759 510.838,97.5199 518.558,109.745 526.278,123.67 533.997,139.617 541.717,158.041 549.436,179.625 557.156,205.482 564.876,237.624 
 572.595,280.316 580.315,343.12 
 "/>
<polyline clip-path="url(#clip7902)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.811 39.9417,358.965 47.6613,333.642 55.381,306.825 63.1006,280.723 70.8202,256.229 78.5398,233.672 86.2594,213.116 93.979,194.501 101.699,177.708 
 109.418,162.597 117.138,149.026 124.858,136.858 132.577,125.964 140.297,116.228 148.016,107.545 155.736,99.8209 163.456,92.9721 171.175,86.9238 178.895,81.6094 
 186.614,76.9698 194.334,72.9523 202.054,69.51 209.773,66.601 217.493,64.188 225.213,62.2376 232.932,60.7199 240.652,59.6083 248.371,58.8787 256.091,58.51 
 263.811,58.4829 271.53,58.7803 279.25,59.387 286.969,60.2894 294.689,61.4754 302.409,62.9345 310.128,64.6572 317.848,66.6355 325.568,68.8623 333.287,71.3319 
 341.007,74.0392 348.726,76.9806 356.446,80.153 364.166,83.5547 371.885,87.1846 379.605,91.0429 387.325,95.1307 395.044,99.4502 402.764,104.005 410.483,108.799 
 418.203,113.839 425.923,119.132 433.642,124.687 441.362,130.516 449.081,136.632 456.801,143.051 464.521,149.794 472.24,156.883 479.96,164.348 487.68,172.223 
 495.399,180.552 503.119,189.389 510.838,198.802 518.558,208.881 526.278,219.746 533.997,231.56 541.717,244.559 549.436,259.104 557.156,275.787 564.876,295.693 
 572.595,321.141 580.315,357.329 
 "/>
<polyline clip-path="url(#clip7902)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,372.171 47.6613,365.957 55.381,358.464 63.1006,350.185 70.8202,341.396 78.5398,332.265 86.2594,322.903 93.979,313.385 101.699,303.766 
 109.418,294.083 117.138,284.366 124.858,274.638 132.577,264.917 140.297,255.219 148.016,245.557 155.736,235.942 163.456,226.385 171.175,216.897 178.895,207.486 
 186.614,198.162 194.334,188.933 202.054,179.809 209.773,170.799 217.493,161.912 225.213,153.156 232.932,144.542 240.652,136.078 248.371,127.776 256.091,119.644 
 263.811,111.694 271.53,103.937 279.25,96.3837 286.969,89.0464 294.689,81.9373 302.409,75.0694 310.128,68.4563 317.848,62.112 325.568,56.0516 333.287,50.2908 
 341.007,44.8461 348.726,39.7352 356.446,34.9768 364.166,30.5905 371.885,26.5977 379.605,23.021 387.325,19.8848 395.044,17.2152 402.764,15.0408 410.483,13.3922 
 418.203,12.3033 425.923,11.811 433.642,11.9561 441.362,12.7838 449.081,14.3448 456.801,16.6961 464.521,19.9023 472.24,24.0377 479.96,29.1883 487.68,35.455 
 495.399,42.958 503.119,51.8425 510.838,62.2878 518.558,74.5205 526.278,88.8347 533.997,105.626 541.717,125.453 549.436,149.146 557.

In [222]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_LS,A11_LS)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_ten,A11_ten)*abs(bw(s,mKs^2,ΓKs))^2
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.01:s_en, lab=["JPAC","LS","tensor"],
    xlab="MpiK(GeV)", ylab="Differentian cross section",
    title="no Blatt-Weisskopf factors",
    legend=:topleft)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 no Blatt-Weisskopf factors 
 
 
 MpiK(GeV) 
 
 
 Differentian cross section 
 
<polyline clip-path="url(#clip5102)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 47.9701,360.065 50.4577,359.12 52.9453,357.466 55.4329,355.402 57.9205,353.046 60.4081,350.47 62.8957,347.723 65.3833,344.843 67.8709,341.859 70.3585,338.794 
 72.8461,335.666 75.3337,332.492 77.8212,329.284 80.3088,326.053 82.7964,322.808 85.284,319.558 87.7716,316.308 90.2592,313.064 92.7468,309.831 95.2344,306.613 
 97.722,303.413 100.21,300.235 102.697,297.079 105.185,293.95 107.672,290.847 110.16,287.772 112.648,284.726 115.135,281.711 117.623,278.726 120.11,275.773 
 122.598,272.85 125.086,269.96 127.573,267.101 130.061,264.273 132.548,261.477 135.036,258.711 137.523,255.976 140.011,253.272 142.499,250.598 144.986,247.952 
 147.474,245.336 149.961,242.748 152.449,240.188 154.937,237.655 157.424,235.148 159.912,232.667 162.399,230.212 164.887,227.781 167.375,225.373 169.862,222.989 
 172.35,220.627 174.837,218.286 177.325,215.967 179.813,213.668 182.3,211.388 184.788,209.127 187.275,206.884 189.763,204.658 192.251,202.449 194.738,200.255 
 197.226,198.077 199.713,195.913 202.201,193.763 204.688,191.625 207.176,189.5 209.664,187.386 212.151,185.283 214.639,183.19 217.126,181.107 219.614,179.031 
 222.102,176.964 224.589,174.904 227.077,172.851 229.564,170.804 232.052,168.761 234.54,166.723 237.027,164.689 239.515,162.659 242.002,160.63 244.49,158.604 
 246.978,156.578 249.465,154.553 251.953,152.529 254.44,150.503 256.928,148.477 259.416,146.448 261.903,144.417 264.391,142.383 266.878,140.346 269.366,138.305 
 271.853,136.259 274.341,134.209 276.829,132.153 279.316,130.092 281.804,128.024 284.291,125.95 286.779,123.87 289.267,121.783 291.754,119.689 294.242,117.588 
 296.729,115.48 299.217,113.365 301.705,111.244 304.192,109.116 306.68,106.982 309.167,104.843 311.655,102.699 314.143,100.551 316.63,98.3995 319.118,96.2466 
 321.605,94.0932 324.093,91.9409 326.581,89.7917 329.068,87.6476 331.556,85.511 334.043,83.3848 336.531,81.272 339.018,79.1761 341.506,77.1012 343.994,75.0516 
 346.481,73.0322 348.969,71.0486 351.456,69.1067 353.944,67.2134 356.432,65.3759 358.919,63.6026 361.407,61.9022 363.894,60.2845 366.382,58.7602 368.87,57.3409 
 371.357,56.0389 373.845,54.8678 376.332,53.8421 378.82,52.9773 381.308,52.2898 383.795,51.7971 386.283,51.5175 388.77,51.4704 391.258,51.6756 393.746,52.1537 
 396.233,52.9257 398.721,54.0127 401.208,55.4358 403.696,57.2157 406.183,59.3723 408.671,61.9244 411.159,64.8892 413.646,68.2819 416.134,72.1149 418.621,76.3977 
 421.109,81.1361 423.597,86.3319 426.084,91.9823 428.572,98.0796 431.059,104.611 433.547,111.558 436.035,118.898 438.522,126.601 441.01,134.635 443.497,142.959 
 445.985,151.533 448.473,160.31 450.96,169.242 453.448,178.28 455.935,187.372 458.423,196.469 460.911,205.52 463.398,214.48 465.886,223.304 468.373,231.95 
 470.861,240.381 473.348,248.564 475.836,256.472 478.324,264.08 480.811,271.369 483.299,278.324 485.786,284.937 488.274,291.199 490.762,297.109 493.249,302.667 
 495.737,307.878 498.224,312.747 500.712,317.283 503.2,321.495 505.687,325.397 508.175,328.999 510.662,332.316 513.15,335.361 515.638,338.149 518.125,340.695 
 520.613,343.012 523.1,345.115 525.588,347.017 528.076,348.733 530.563,350.276 533.051,351.657 535.538,352.889 538.026,353.984 540.513,354.951 543.001,355.803 
 545.489,356.547 547.976,357.195 550.464,357.753 552.951,358.231 555.439,358.636 557.927,358.975 560.414,359.255 562.902,359.482 565.389,359.662 567.877,359.802 
 570.365,359.906 572.852,359.979 575.34,360.026 577.827,360.053 580.315,360.064 
 "/>
<polyline clip-path="url(#clip5102)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 47.9701,360.063 50.4577,3

In [184]:
savefig("/tmp/noBW.pdf")

In [185]:
BWs = [z->1./(1.+z), z->1./(9+3*z+z^2)]
Rbw=5; # /GeV
p(s) = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
q(s) = sqrt(λ(s,mπ^2,mK^2)/(4*s))
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*abs(bw(s,mKs^2,ΓKs))^2*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2),
    s->project_general(s,0,1,A10_LS,A11_LS)*abs(bw(s,mKs^2,ΓKs))^2*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2),
    s->project_general(s,0,1,A10_ten,A11_ten)*abs(bw(s,mKs^2,ΓKs))^2*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2) # ,
#    s->abs(bw(s,mKs^2,ΓKs))^2
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.01:s_en, lab=["JPAC","LS","tensor"],
    xlab="MpiK(GeV)", ylab="Differentian cross section",
    title="with Blatt-Weisskopf factors",
    legend=:topleft)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 1.25 
 
 
 with Blatt-Weisskopf factors 
 
 
 MpiK(GeV) 
 
 
 Differentian cross section 
 
<polyline clip-path="url(#clip1302)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 54.5598,360.065 57.0166,360.028 59.4734,359.96 61.9302,359.873 64.387,359.771 66.8438,359.655 69.3006,359.529 71.7574,359.393 74.2142,359.248 76.671,359.094 
 79.1278,358.933 81.5846,358.764 84.0414,358.588 86.4982,358.405 88.955,358.216 91.4118,358.02 93.8686,357.818 96.3254,357.61 98.7822,357.396 101.239,357.176 
 103.696,356.95 106.153,356.717 108.609,356.479 111.066,356.234 113.523,355.983 115.98,355.726 118.437,355.463 120.893,355.193 123.35,354.917 125.807,354.634 
 128.264,354.344 130.721,354.047 133.177,353.744 135.634,353.433 138.091,353.114 140.548,352.788 143.005,352.454 145.461,352.113 147.918,351.763 150.375,351.404 
 152.832,351.037 155.289,350.662 157.745,350.277 160.202,349.883 162.659,349.479 165.116,349.065 167.573,348.641 170.029,348.207 172.486,347.762 174.943,347.305 
 177.4,346.838 179.857,346.358 182.313,345.867 184.77,345.362 187.227,344.845 189.684,344.315 192.141,343.77 194.597,343.212 197.054,342.639 199.511,342.05 
 201.968,341.447 204.425,340.826 206.881,340.19 209.338,339.535 211.795,338.863 214.252,338.173 216.709,337.464 219.165,336.734 221.622,335.985 224.079,335.214 
 226.536,334.421 228.993,333.606 231.449,332.767 233.906,331.904 236.363,331.016 238.82,330.101 241.277,329.16 243.733,328.19 246.19,327.192 248.647,326.163 
 251.104,325.103 253.561,324.011 256.017,322.885 258.474,321.724 260.931,320.527 263.388,319.293 265.845,318.019 268.301,316.705 270.758,315.348 273.215,313.948 
 275.672,312.503 278.129,311.01 280.585,309.468 283.042,307.875 285.499,306.229 287.956,304.528 290.413,302.77 292.869,300.952 295.326,299.072 297.783,297.128 
 300.24,295.116 302.697,293.036 305.153,290.883 307.61,288.655 310.067,286.349 312.524,283.961 314.981,281.49 317.437,278.931 319.894,276.281 322.351,273.537 
 324.808,270.695 327.265,267.752 329.721,264.704 332.178,261.547 334.635,258.277 337.092,254.892 339.549,251.386 342.005,247.757 344.462,244 346.919,240.111 
 349.376,236.088 351.833,231.928 354.289,227.626 356.746,223.179 359.203,218.587 361.66,213.845 364.117,208.954 366.573,203.911 369.03,198.717 371.487,193.372 
 373.944,187.878 376.401,182.237 378.857,176.454 381.314,170.533 383.771,164.482 386.228,158.31 388.685,152.027 391.141,145.647 393.598,139.184 396.055,132.657 
 398.512,126.086 400.969,119.494 403.425,112.909 405.882,106.358 408.339,99.8746 410.796,93.4937 413.253,87.253 415.709,81.193 418.166,75.3563 420.623,69.7872 
 423.08,64.5311 425.537,59.6341 427.993,55.142 430.45,51.0997 432.907,47.5502 435.364,44.5335 437.821,42.0861 440.277,40.2398 442.734,39.021 445.191,38.4498 
 447.648,38.5399 450.105,39.2977 452.561,40.7221 455.018,42.8051 457.475,45.5311 459.932,48.8781 462.389,52.8176 464.845,57.3159 467.302,62.3344 469.759,67.8308 
 472.216,73.7601 474.673,80.0753 477.129,86.7287 479.586,93.6722 482.043,100.859 484.5,108.243 486.957,115.78 489.413,123.429 491.87,131.153 494.327,138.914 
 496.784,146.681 499.241,154.426 501.697,162.122 504.154,169.747 506.611,177.283 509.068,184.712 511.525,192.021 513.981,199.2 516.438,206.241 518.895,213.137 
 521.352,219.884 523.809,226.48 526.265,232.925 528.722,239.22 531.179,245.367 533.636,251.37 536.093,257.236 538.549,262.969 541.006,268.579 543.463,274.073 
 545.92,279.463 548.377,284.76 550.833,289.979 553.29,295.136 555.747,300.25 558.204,305.345 560.661,310.448 563.117,315.596 565.574,320.836 568.031,326.227 
 570.488,331.853 572.945,337.821 575.401,344.256 577.858,351.181 580.315,357.766 
 "/>
<polyline clip-path="url(#clip1302)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 54.5598,360.065 57.0166,

In [186]:
savefig("/tmp/withBW.pdf")

In [225]:
adam_data = readdlm("fort.13");

In [243]:
normB = sum(adam_data[:,2])
dx = adam_data[2,1]-adam_data[1,1]
bar(adam_data[:,1],adam_data[:,2]/normB/dx, lab="Adam")
plot!(nfuncs[3], s_th:0.01:s_en, lab="Misha", c=:black,
    xlab="MpiK(GeV)", ylab="Differentian cross section",
    title="no Blatt-Weisskopf factors",
    legend=:topleft)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.00 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 no Blatt-Weisskopf factors 
 
 
 MpiK(GeV) 
 
 
 Differentian cross section 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
<polyline clip-path="url(#clip7502)" style="stroke:#000000; stroke-width:1; stroke-opacity:1; fill:none" points="
 65.8306,360.065 68.161,359.367 70.4913,358.142 72.8217,356.605 75.1521,354.843 77.4824,352.905 79.8128,350.829 82.1431,348.64 84.4735,346.361 86.8038,344.007 
 89.1342,341.593 91.4645,339.13 93.7949,336.628 96.1253,334.094 98.4556,331.536 100.786,328.959 103.116,326.368 105.447,323.769 107.777,321.163 110.107,318.556 
 112.438,315.948 114.768,313.344 117.098,310.743 119.429,308.15 121.759,305.564 124.09,302.986 126.42,300.419 128.75,297.863 131.081,295.318 133.411,292.785 
 135.741,290.265 138.072,287.757 140.402,285.263 142.732,282.782 145.063,280.313 147.393,277.859 149.723,275.417 152.054,272.988 154.384,270.572 156.714,268.169 
 159.045,265.779 161.375,263.401 163.706,261.034 166.036,258.68 168.366,256.336 170.697,254.004 173.027,251.682 175.357,249.37 177.688,247.067 180.018,244.775 
 182.348,242.49 184.679,240.215 187.009,237.947 189.339,235.686 191.67,233.433 194,231.186 196.331,228.945 198.661,226.71 200.991,224.479 203.322,222.253 
 205.652,220.031 207.982,217.812 210.313,215.596 212.643,213.382 214.973,211.17 217.304,208.96 219.634,206.75 221.964,204.54 224.295,202.33 226.625,200.119 
 228.956,197.906 231.286,195.691 233.616,193.474 235.947,191.253 238.277,189.029 240.607,186.801 242.938,184.567 245.268,182.328 247.598,180.083 249.929,177.832 
 252.259,175.573 254.589,173.306 256.92,171.032 259.25,168.748 261.58,166.456 263.911,164.153 266.241,161.84 268.572,159.516 270.902,157.181 273.232,154.834 
 275.563,152.474 277.893,150.102 280.223,147.717 282.554,145.318 284.884,142.905 287.214,140.478 289.545,138.037 291.875,135.581 294.205,133.11 296.536,130.625 
 298.866,128.124 301.197,125.608 303.527,123.078 305.857,120.533 308.188,117.975 310.518,115.402 312.848,112.816 315.179,110.219 317.509,107.609 319.839,104.99 
 322.17,102.361 324.5,99.7257 326.83,97.0844 329.161,94.4398 331.491,91.7942 333.822,89.1504 336.152,86.5115 338.482,83.8811 340.813,81.2631 343.143,78.6621 
 345.473,76.0831 347.804,73.5318 350.134,71.0144 352.464,68.5378 354.795,66.1098 357.125,63.7388 359.455,61.4341 361.786,59.2061 364.116,57.0659 366.446,55.0257 
 368.777,53.0989 371.107,51.2997 373.438,49.6436 375.768,48.1473 378.098,46.8285 380.429,45.7061 382.759,44.8 385.089,44.1312 387.42,43.7217 389.75,43.5939 
 392.08,43.7712 394.411,44.2772 396.741,45.1355 399.071,46.3696 401.402,48.0023 403.732,50.0554 406.063,52.5492 408.393,55.5019 410.723,58.9291 413.054,62.8434 
 415.384,67.2535 417.714,72.164 420.045,77.5746 422.375,83.48 424.705,89.8691 427.036,96.7254 429.366,104.026 431.696,111.743 434.027,119.842 436.357,128.285 
 438.688,137.027 441.018,146.021 443.348,155.217 445.679,164.561 448.009,174.001 450.339,183.481 452.67,192.948 455,202.352 457.33,211.643 459.661,220.775 
 461.991,229.708 464.321,238.404 466.652,246.83 468.982,254.959 471.312,262.768 473.643,270.239 475.973,277.36 478.304,284.12 480.634,290.515 482.964,296.545 
 485.295,302.21 487.625,307.517 489.955,312.471 492.286,317.083 494.616,321.364 496.946,325.325 499.277,328.981 501.607,332.345 503.937,335.431 506.268,338.255 
 508.598,340.831 510.929,343.174 513.259,345.299 515.589,347.219 517.92,348.949 520.25,350.501 522.58,351.889 524.911,353.124 527.241,354.219 529.571,355.184 
 531.902,356.03 534.232,356.767 536.562,357.404 538.893,357.95 541.223,358.414 543.553,358.803 545.884,359.125 548.214,359.388 550.545,359.596 552.875,359.758 
 555.205,359.878 557.536,359.963 

In [244]:
savefig("/tmp/adam_tensor.pdf")